{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(476, 449)\n",
      "(476, 133)\n",
      "(476, 14875)\n",
      "(1552, 2)\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os \n",
    "from scipy.stats import spearmanr, pearsonr\n",
    "from statsmodels.stats.multitest import multipletests\n",
    "from scipy import stats\n",
    "\n",
    "path = os.path.join(os.getcwd(), \"data\")\n",
    "output_path = os.path.join(os.getcwd(),\"analysis_output\")\n",
    "tax_data = pd.read_csv(path+\"/PC_normalized_PC92_HC384_130 genus_ML.csv\")\n",
    "path_data = pd.read_csv(path+\"/PC Pathway.csv\")\n",
    "ortho_data = pd.read_csv(path+'/PC Orthology.csv')\n",
    "mapping_df = pd.read_excel(path+ \"/pathway_Orthology_map.xlsx\")\n",
    "ortho_label = pd.read_excel(path+'/orthology_label_list.xlsx')\n",
    "tax_data\n",
    "print(path_data.shape)\n",
    "print(tax_data.shape)\n",
    "print(ortho_data.shape)\n",
    "print(mapping_df.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_no</th>\n",
       "      <th>group</th>\n",
       "      <th>group_1</th>\n",
       "      <th>Abiotrophia</th>\n",
       "      <th>Acetatifactor</th>\n",
       "      <th>Acholeplasmataceae_uc</th>\n",
       "      <th>Actinobacillus</th>\n",
       "      <th>Actinobaculum</th>\n",
       "      <th>Actinomyces</th>\n",
       "      <th>Aggregatibacter</th>\n",
       "      <th>...</th>\n",
       "      <th>Xanthomonas</th>\n",
       "      <th>Cloacibacterium</th>\n",
       "      <th>Legionella</th>\n",
       "      <th>Variovorax</th>\n",
       "      <th>Bergeriella</th>\n",
       "      <th>Ethanoligenens</th>\n",
       "      <th>Lentimicrobiaceae_uc</th>\n",
       "      <th>Moraxella</th>\n",
       "      <th>Mycoplasma_g13</th>\n",
       "      <th>Streptobacillus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>H1700097</td>\n",
       "      <td>Cancer</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3.70</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>H1700105</td>\n",
       "      <td>Cancer</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.70</td>\n",
       "      <td>0.90</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>H1700084</td>\n",
       "      <td>Cancer</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.30</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>H1700109</td>\n",
       "      <td>Cancer</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.80</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>H1700098</td>\n",
       "      <td>Cancer</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2.50</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>471</th>\n",
       "      <td>H1900931</td>\n",
       "      <td>Control</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.94</td>\n",
       "      <td>0.68</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>472</th>\n",
       "      <td>H1900932</td>\n",
       "      <td>Control</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.94</td>\n",
       "      <td>0.27</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>473</th>\n",
       "      <td>H1900933</td>\n",
       "      <td>Control</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.45</td>\n",
       "      <td>0.93</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>474</th>\n",
       "      <td>H1900936</td>\n",
       "      <td>Control</td>\n",
       "      <td>0</td>\n",
       "      <td>0.37</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.82</td>\n",
       "      <td>0.03</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475</th>\n",
       "      <td>H1900939</td>\n",
       "      <td>Control</td>\n",
       "      <td>0</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2.14</td>\n",
       "      <td>0.03</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>476 rows × 133 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     study_no    group  group_1  Abiotrophia  Acetatifactor  \\\n",
       "0    H1700097   Cancer        1         0.00            0.0   \n",
       "1    H1700105   Cancer        1         0.00            0.0   \n",
       "2    H1700084   Cancer        1         0.00            0.0   \n",
       "3    H1700109   Cancer        1         0.00            0.0   \n",
       "4    H1700098   Cancer        1         0.00            0.0   \n",
       "..        ...      ...      ...          ...            ...   \n",
       "471  H1900931  Control        0         0.00            0.0   \n",
       "472  H1900932  Control        0         0.00            0.0   \n",
       "473  H1900933  Control        0         0.00            0.0   \n",
       "474  H1900936  Control        0         0.37            0.0   \n",
       "475  H1900939  Control        0         0.33            0.0   \n",
       "\n",
       "     Acholeplasmataceae_uc  Actinobacillus  Actinobaculum  Actinomyces  \\\n",
       "0                      0.0            0.00           0.00         3.70   \n",
       "1                      0.0            0.00           0.00         0.70   \n",
       "2                      0.0            0.00           0.00         1.30   \n",
       "3                      0.0            0.00           0.10         0.80   \n",
       "4                      0.0            0.00           0.00         2.50   \n",
       "..                     ...             ...            ...          ...   \n",
       "471                    0.0            0.03           0.06         0.94   \n",
       "472                    0.0            0.00           0.04         0.94   \n",
       "473                    0.0            0.00           0.02         0.45   \n",
       "474                    0.0            0.00           0.00         1.82   \n",
       "475                    0.0            0.00           0.00         2.14   \n",
       "\n",
       "     Aggregatibacter  ...  Xanthomonas  Cloacibacterium  Legionella  \\\n",
       "0               0.00  ...          0.0              0.0         0.0   \n",
       "1               0.90  ...          0.0              0.0         0.0   \n",
       "2               0.00  ...          0.0              0.0         0.0   \n",
       "3               0.00  ...          0.0              0.2         0.0   \n",
       "4               0.00  ...          0.0              0.0         0.0   \n",
       "..               ...  ...          ...              ...         ...   \n",
       "471             0.68  ...          0.0              0.0         0.0   \n",
       "472             0.27  ...          0.0              0.0         0.0   \n",
       "473             0.93  ...          0.0              0.0         0.0   \n",
       "474             0.03  ...          0.0              0.0         0.0   \n",
       "475             0.03  ...          0.0              0.0         0.0   \n",
       "\n",
       "     Variovorax  Bergeriella  Ethanoligenens  Lentimicrobiaceae_uc  Moraxella  \\\n",
       "0           0.0          0.0             0.0                   0.0        0.0   \n",
       "1           0.0          0.0             0.0                   0.0        0.0   \n",
       "2           0.0          0.0             0.0                   0.0        0.0   \n",
       "3           0.0          0.0             0.0                   0.0        0.0   \n",
       "4           0.0          0.0             0.0                   0.0        0.0   \n",
       "..          ...          ...             ...                   ...        ...   \n",
       "471         0.0          0.0             0.0                   0.0        0.0   \n",
       "472         0.0          0.0             0.0                   0.0        0.0   \n",
       "473         0.0          0.0             0.0                   0.0        0.0   \n",
       "474         0.0          0.0             0.0                   0.0        0.0   \n",
       "475         0.0          0.0             0.0                   0.0        0.0   \n",
       "\n",
       "     Mycoplasma_g13  Streptobacillus  \n",
       "0               0.0              0.0  \n",
       "1               0.0              0.0  \n",
       "2               0.0              0.0  \n",
       "3               0.0              0.0  \n",
       "4               0.0              0.0  \n",
       "..              ...              ...  \n",
       "471             0.0              0.0  \n",
       "472             0.0              0.0  \n",
       "473             0.0              0.0  \n",
       "474             0.0              0.0  \n",
       "475             0.0              0.0  \n",
       "\n",
       "[476 rows x 133 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tax_data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "데이터 구조 확인"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set 'study_no' as the index for each DataFrame\n",
    "tax_data.set_index('study_no', inplace=True)\n",
    "path_data.set_index('study_no', inplace=True)\n",
    "ortho_data.set_index('study_no', inplace=True)\n",
    "\n",
    "# Sort each DataFrame by the index in ascending order\n",
    "tax_data.sort_index(inplace=True)\n",
    "path_data.sort_index(inplace=True)\n",
    "ortho_data.sort_index(inplace=True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 기존 Taxonomy Analysis 결과에 따른 genus 선정"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Up-regulated 된 genus와 down-regulated 된 genus 데이터 추출"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Desulfovibrio</th>\n",
       "      <th>Fretibacterium</th>\n",
       "      <th>Lactobacillus</th>\n",
       "      <th>Leuconostoc</th>\n",
       "      <th>Olsenella</th>\n",
       "      <th>Parvimonas</th>\n",
       "      <th>Ralstonia</th>\n",
       "      <th>Pseudomonas</th>\n",
       "      <th>Simonsiella</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>study_no</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>H1700001</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.48</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>1.71</td>\n",
       "      <td>9.06</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1700002</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1700003</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1700004</th>\n",
       "      <td>0.04</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.35</td>\n",
       "      <td>6.24</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1700005</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1700106</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.60</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.40</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1700107</th>\n",
       "      <td>0.00</td>\n",
       "      <td>2.10</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.50</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1700108</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1700109</th>\n",
       "      <td>0.30</td>\n",
       "      <td>3.20</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.50</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1700110</th>\n",
       "      <td>0.60</td>\n",
       "      <td>1.10</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.70</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>92 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Desulfovibrio  Fretibacterium  Lactobacillus  Leuconostoc  \\\n",
       "study_no                                                              \n",
       "H1700001           0.00            0.48           0.00         0.03   \n",
       "H1700002           0.00            0.00           0.05         0.00   \n",
       "H1700003           0.00            0.00           0.02         0.00   \n",
       "H1700004           0.04            0.24           0.23         0.00   \n",
       "H1700005           0.00            0.15           0.24         0.00   \n",
       "...                 ...             ...            ...          ...   \n",
       "H1700106           0.00            0.60           0.00         0.00   \n",
       "H1700107           0.00            2.10           0.00         0.00   \n",
       "H1700108           0.00            0.00           0.00         0.00   \n",
       "H1700109           0.30            3.20           0.00         0.00   \n",
       "H1700110           0.60            1.10           0.00         0.00   \n",
       "\n",
       "          Olsenella  Parvimonas  Ralstonia  Pseudomonas  Simonsiella  \n",
       "study_no                                                              \n",
       "H1700001       1.71        9.06       0.00          0.0          0.0  \n",
       "H1700002       0.02        0.19       0.00          0.0          0.0  \n",
       "H1700003       0.00        0.00       0.00          0.0          0.0  \n",
       "H1700004       0.35        6.24       0.00          0.0          0.0  \n",
       "H1700005       0.03        0.13       0.02          0.0          0.0  \n",
       "...             ...         ...        ...          ...          ...  \n",
       "H1700106       0.00        0.40       0.00          0.0          0.0  \n",
       "H1700107       0.00        0.50       0.00          0.0          0.0  \n",
       "H1700108       0.00        0.10       0.00          0.0          0.0  \n",
       "H1700109       0.00        0.10       0.50          0.0          0.0  \n",
       "H1700110       0.00        0.20       0.70          0.0          0.0  \n",
       "\n",
       "[92 rows x 9 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "use_genus = ['group_1','Desulfovibrio', 'Fretibacterium', 'Lactobacillus', 'Leuconostoc', \n",
    "                      'Olsenella', 'Parvimonas', 'Ralstonia','Pseudomonas', 'Simonsiella']\n",
    "use_genus_data = tax_data[use_genus]\n",
    "pc_tax_data = use_genus_data[use_genus_data['group_1'] == 1].drop('group_1', axis=1)\n",
    "control_tax_data = use_genus_data[use_genus_data['group_1'] == 0].drop('group_1', axis=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Pathway 통계 분석을 통해 선정된 up-regulated된 pathway와 down-regulated된 pathway 선정"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "path_data = pd.read_excel(output_path+'/statistical_analysis_results_with_labels_pathway.xlsx',index_col = 0)\n",
    "ortho_data = pd.read_excel(output_path+'/statistical_analysis_results_with_labels_Orthology.xlsx',index_col = 0)\n",
    "data = path_data.dropna(subset=['log2FC_value', 'Wilcoxon_p_value(-log10)'])\n",
    "data2 = ortho_data.dropna(subset = ['log2FC_value', 'Wilcoxon_p_value(-log10)'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "fold_change_threshold = 1.0  # threshold for log2 fold change\n",
    "significance_threshold = 1.3  # threshold for -log10(p-value) (~p < 0.05)\n",
    "logistic_threshold = 0.05  # threshold for logistic Regression with false discovery rate adjusted p-value\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(476, 9) (476, 15)\n"
     ]
    }
   ],
   "source": [
    "\n",
    "# Based on statistical analysis from pathway bring out upregulated pathway in Cancer group and downregulated pathway in Cancer group\n",
    "# Correcting the formatting of down_path and up_path lists\n",
    "up_path = data.loc[(data['log2FC_value'] > fold_change_threshold) & \n",
    "         (data['Wilcoxon_p_value(-log10)'] > significance_threshold) &\n",
    "         (data['LogReg_p_fdr'] < logistic_threshold), 'color'] = 'red'\n",
    "down_path = data.loc[(data['log2FC_value'] < -fold_change_threshold) & \n",
    "         (data['Wilcoxon_p_value(-log10)'] > significance_threshold) &\n",
    "         (data['LogReg_p_fdr'] < logistic_threshold),\n",
    "up_path = [\n",
    "    'ko04024', 'ko04977', 'ko00601', 'ko04530', 'ko04151',\n",
    "    'ko04141', 'ko01053', 'ko00220', 'ko04120'\n",
    "]\n",
    "use_path ='[ group_2',''\n",
    "# Filtering the path_data DataFrame for the upregulated and downregulated pathways\n",
    "up_path_data = path_data[up_path]\n",
    "down_path_data = path_data[down_path]\n",
    "print(up_path_data.shape, down_path_data.shape)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ko04024</th>\n",
       "      <th>ko04977</th>\n",
       "      <th>ko00601</th>\n",
       "      <th>ko04530</th>\n",
       "      <th>ko04151</th>\n",
       "      <th>ko04141</th>\n",
       "      <th>ko01053</th>\n",
       "      <th>ko00220</th>\n",
       "      <th>ko04120</th>\n",
       "      <th>ko00472</th>\n",
       "      <th>...</th>\n",
       "      <th>ko03320</th>\n",
       "      <th>ko00333</th>\n",
       "      <th>ko00471</th>\n",
       "      <th>ko04750</th>\n",
       "      <th>ko00830</th>\n",
       "      <th>ko00944</th>\n",
       "      <th>ko00404</th>\n",
       "      <th>ko04330</th>\n",
       "      <th>ko00565</th>\n",
       "      <th>ko04975</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>study_no</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>H1700001</th>\n",
       "      <td>0.000321</td>\n",
       "      <td>0.000299</td>\n",
       "      <td>0.000184</td>\n",
       "      <td>0.000992</td>\n",
       "      <td>0.001175</td>\n",
       "      <td>0.000683</td>\n",
       "      <td>0.000343</td>\n",
       "      <td>0.003190</td>\n",
       "      <td>0.000468</td>\n",
       "      <td>0.000154</td>\n",
       "      <td>...</td>\n",
       "      <td>0.001280</td>\n",
       "      <td>0.001200</td>\n",
       "      <td>0.001452</td>\n",
       "      <td>0.000199</td>\n",
       "      <td>0.000728</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>0.000347</td>\n",
       "      <td>0.000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1700002</th>\n",
       "      <td>0.000228</td>\n",
       "      <td>0.000114</td>\n",
       "      <td>0.000115</td>\n",
       "      <td>0.000411</td>\n",
       "      <td>0.000665</td>\n",
       "      <td>0.000561</td>\n",
       "      <td>0.000465</td>\n",
       "      <td>0.003634</td>\n",
       "      <td>0.000304</td>\n",
       "      <td>0.000111</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000905</td>\n",
       "      <td>0.001383</td>\n",
       "      <td>0.001380</td>\n",
       "      <td>0.000179</td>\n",
       "      <td>0.000539</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>0.000064</td>\n",
       "      <td>0.000193</td>\n",
       "      <td>0.000009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1700003</th>\n",
       "      <td>0.000772</td>\n",
       "      <td>0.000160</td>\n",
       "      <td>0.000055</td>\n",
       "      <td>0.000402</td>\n",
       "      <td>0.000877</td>\n",
       "      <td>0.000403</td>\n",
       "      <td>0.000836</td>\n",
       "      <td>0.004040</td>\n",
       "      <td>0.000826</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000846</td>\n",
       "      <td>0.001387</td>\n",
       "      <td>0.001205</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.000722</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>0.000105</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0.000468</td>\n",
       "      <td>0.000007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1700004</th>\n",
       "      <td>0.000159</td>\n",
       "      <td>0.000171</td>\n",
       "      <td>0.000093</td>\n",
       "      <td>0.000362</td>\n",
       "      <td>0.000662</td>\n",
       "      <td>0.000578</td>\n",
       "      <td>0.000480</td>\n",
       "      <td>0.003526</td>\n",
       "      <td>0.000244</td>\n",
       "      <td>0.000136</td>\n",
       "      <td>...</td>\n",
       "      <td>0.001106</td>\n",
       "      <td>0.001286</td>\n",
       "      <td>0.001492</td>\n",
       "      <td>0.000134</td>\n",
       "      <td>0.000530</td>\n",
       "      <td>0.000019</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>0.000033</td>\n",
       "      <td>0.000156</td>\n",
       "      <td>0.000019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1700005</th>\n",
       "      <td>0.000202</td>\n",
       "      <td>0.000112</td>\n",
       "      <td>0.000138</td>\n",
       "      <td>0.000334</td>\n",
       "      <td>0.000645</td>\n",
       "      <td>0.000787</td>\n",
       "      <td>0.000502</td>\n",
       "      <td>0.003485</td>\n",
       "      <td>0.000328</td>\n",
       "      <td>0.000090</td>\n",
       "      <td>...</td>\n",
       "      <td>0.001230</td>\n",
       "      <td>0.001375</td>\n",
       "      <td>0.001387</td>\n",
       "      <td>0.000153</td>\n",
       "      <td>0.000357</td>\n",
       "      <td>0.000019</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0.000047</td>\n",
       "      <td>0.000135</td>\n",
       "      <td>0.000013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1900931</th>\n",
       "      <td>0.000019</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.000126</td>\n",
       "      <td>0.000190</td>\n",
       "      <td>0.000258</td>\n",
       "      <td>0.000243</td>\n",
       "      <td>0.001628</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>0.003436</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003813</td>\n",
       "      <td>0.004632</td>\n",
       "      <td>0.004155</td>\n",
       "      <td>0.000743</td>\n",
       "      <td>0.001940</td>\n",
       "      <td>0.001301</td>\n",
       "      <td>0.000717</td>\n",
       "      <td>0.002465</td>\n",
       "      <td>0.003482</td>\n",
       "      <td>0.000257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1900932</th>\n",
       "      <td>0.000015</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.000256</td>\n",
       "      <td>0.000229</td>\n",
       "      <td>0.000279</td>\n",
       "      <td>0.001748</td>\n",
       "      <td>0.000292</td>\n",
       "      <td>0.003323</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003710</td>\n",
       "      <td>0.004256</td>\n",
       "      <td>0.003987</td>\n",
       "      <td>0.000836</td>\n",
       "      <td>0.001826</td>\n",
       "      <td>0.001867</td>\n",
       "      <td>0.000796</td>\n",
       "      <td>0.002764</td>\n",
       "      <td>0.003162</td>\n",
       "      <td>0.000270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1900933</th>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>0.000023</td>\n",
       "      <td>0.000118</td>\n",
       "      <td>0.000222</td>\n",
       "      <td>0.000256</td>\n",
       "      <td>0.000246</td>\n",
       "      <td>0.001847</td>\n",
       "      <td>0.000279</td>\n",
       "      <td>0.003322</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003726</td>\n",
       "      <td>0.004564</td>\n",
       "      <td>0.004190</td>\n",
       "      <td>0.000701</td>\n",
       "      <td>0.001915</td>\n",
       "      <td>0.001220</td>\n",
       "      <td>0.000740</td>\n",
       "      <td>0.002572</td>\n",
       "      <td>0.003535</td>\n",
       "      <td>0.000285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1900936</th>\n",
       "      <td>0.000011</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0.000019</td>\n",
       "      <td>0.000063</td>\n",
       "      <td>0.000091</td>\n",
       "      <td>0.000161</td>\n",
       "      <td>0.000227</td>\n",
       "      <td>0.001817</td>\n",
       "      <td>0.000109</td>\n",
       "      <td>0.003300</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003732</td>\n",
       "      <td>0.005254</td>\n",
       "      <td>0.004255</td>\n",
       "      <td>0.000357</td>\n",
       "      <td>0.002073</td>\n",
       "      <td>0.000565</td>\n",
       "      <td>0.000865</td>\n",
       "      <td>0.002263</td>\n",
       "      <td>0.003407</td>\n",
       "      <td>0.000502</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1900939</th>\n",
       "      <td>0.000015</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.000078</td>\n",
       "      <td>0.000115</td>\n",
       "      <td>0.000195</td>\n",
       "      <td>0.000253</td>\n",
       "      <td>0.001793</td>\n",
       "      <td>0.000130</td>\n",
       "      <td>0.003251</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003749</td>\n",
       "      <td>0.005150</td>\n",
       "      <td>0.004209</td>\n",
       "      <td>0.000420</td>\n",
       "      <td>0.002065</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>0.000830</td>\n",
       "      <td>0.002309</td>\n",
       "      <td>0.003414</td>\n",
       "      <td>0.000479</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>476 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           ko04024   ko04977   ko00601   ko04530   ko04151   ko04141  \\\n",
       "study_no                                                               \n",
       "H1700001  0.000321  0.000299  0.000184  0.000992  0.001175  0.000683   \n",
       "H1700002  0.000228  0.000114  0.000115  0.000411  0.000665  0.000561   \n",
       "H1700003  0.000772  0.000160  0.000055  0.000402  0.000877  0.000403   \n",
       "H1700004  0.000159  0.000171  0.000093  0.000362  0.000662  0.000578   \n",
       "H1700005  0.000202  0.000112  0.000138  0.000334  0.000645  0.000787   \n",
       "...            ...       ...       ...       ...       ...       ...   \n",
       "H1900931  0.000019  0.000015  0.000022  0.000126  0.000190  0.000258   \n",
       "H1900932  0.000015  0.000012  0.000020  0.000174  0.000256  0.000229   \n",
       "H1900933  0.000017  0.000006  0.000023  0.000118  0.000222  0.000256   \n",
       "H1900936  0.000011  0.000010  0.000019  0.000063  0.000091  0.000161   \n",
       "H1900939  0.000015  0.000018  0.000022  0.000078  0.000115  0.000195   \n",
       "\n",
       "           ko01053   ko00220   ko04120   ko00472  ...   ko03320   ko00333  \\\n",
       "study_no                                          ...                       \n",
       "H1700001  0.000343  0.003190  0.000468  0.000154  ...  0.001280  0.001200   \n",
       "H1700002  0.000465  0.003634  0.000304  0.000111  ...  0.000905  0.001383   \n",
       "H1700003  0.000836  0.004040  0.000826  0.000097  ...  0.000846  0.001387   \n",
       "H1700004  0.000480  0.003526  0.000244  0.000136  ...  0.001106  0.001286   \n",
       "H1700005  0.000502  0.003485  0.000328  0.000090  ...  0.001230  0.001375   \n",
       "...            ...       ...       ...       ...  ...       ...       ...   \n",
       "H1900931  0.000243  0.001628  0.000261  0.003436  ...  0.003813  0.004632   \n",
       "H1900932  0.000279  0.001748  0.000292  0.003323  ...  0.003710  0.004256   \n",
       "H1900933  0.000246  0.001847  0.000279  0.003322  ...  0.003726  0.004564   \n",
       "H1900936  0.000227  0.001817  0.000109  0.003300  ...  0.003732  0.005254   \n",
       "H1900939  0.000253  0.001793  0.000130  0.003251  ...  0.003749  0.005150   \n",
       "\n",
       "           ko00471   ko04750   ko00830   ko00944   ko00404   ko04330  \\\n",
       "study_no                                                               \n",
       "H1700001  0.001452  0.000199  0.000728  0.000032  0.000008  0.000038   \n",
       "H1700002  0.001380  0.000179  0.000539  0.000013  0.000030  0.000064   \n",
       "H1700003  0.001205  0.000288  0.000722  0.000016  0.000105  0.000010   \n",
       "H1700004  0.001492  0.000134  0.000530  0.000019  0.000002  0.000033   \n",
       "H1700005  0.001387  0.000153  0.000357  0.000019  0.000011  0.000047   \n",
       "...            ...       ...       ...       ...       ...       ...   \n",
       "H1900931  0.004155  0.000743  0.001940  0.001301  0.000717  0.002465   \n",
       "H1900932  0.003987  0.000836  0.001826  0.001867  0.000796  0.002764   \n",
       "H1900933  0.004190  0.000701  0.001915  0.001220  0.000740  0.002572   \n",
       "H1900936  0.004255  0.000357  0.002073  0.000565  0.000865  0.002263   \n",
       "H1900939  0.004209  0.000420  0.002065  0.000693  0.000830  0.002309   \n",
       "\n",
       "           ko00565   ko04975  \n",
       "study_no                      \n",
       "H1700001  0.000347  0.000038  \n",
       "H1700002  0.000193  0.000009  \n",
       "H1700003  0.000468  0.000007  \n",
       "H1700004  0.000156  0.000019  \n",
       "H1700005  0.000135  0.000013  \n",
       "...            ...       ...  \n",
       "H1900931  0.003482  0.000257  \n",
       "H1900932  0.003162  0.000270  \n",
       "H1900933  0.003535  0.000285  \n",
       "H1900936  0.003407  0.000502  \n",
       "H1900939  0.003414  0.000479  \n",
       "\n",
       "[476 rows x 24 columns]"
      ]
     },
     "execution_count": 272,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "selected_path = path_data[up_path+down_path]\n",
    "selected_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from scipy import stats\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from statsmodels.stats.multitest import multipletests\n",
    "\n",
    "# Step 1: Filter taxonomy and pathway data based on selected columns\n",
    "def filter_data(data, selected_columns):\n",
    "    return data[selected_columns]\n",
    "\n",
    "\n",
    "\n",
    "# Step 2: Check normality of data using Shapiro-Wilk test\n",
    "def check_normality(series, alpha=0.05):\n",
    "    return stats.shapiro(series).pvalue > alpha\n",
    "\n",
    "# Step 3: Compute correlation based on normality\n",
    "def apply_correlation(x, y, alpha=0.05):\n",
    "    # Ensure x and y are 1-dimensional\n",
    "    if x.ndim > 1:\n",
    "        x = x.squeeze()  # Convert to 1D if possible\n",
    "    if y.ndim > 1:\n",
    "        y = y.squeeze()  # Convert to 1D if possible\n",
    "    \n",
    "    # Verify that x and y are now 1D\n",
    "    if x.ndim != 1 or y.ndim != 1:\n",
    "        raise ValueError(\"Both x and y must be 1-dimensional for correlation computation.\")\n",
    "    \n",
    "    # Apply correlation based on normality\n",
    "    if check_normality(x, alpha) and check_normality(y, alpha):\n",
    "        corr, p_value = stats.pearsonr(x, y)\n",
    "        method = \"Pearson\"\n",
    "    else:\n",
    "        corr, p_value = stats.spearmanr(x, y)\n",
    "        method = \"Spearman\"\n",
    "\n",
    "    # Ensure that corr and p_value are scalar values, not matrices\n",
    "    if isinstance(corr, np.ndarray) or isinstance(p_value, np.ndarray):\n",
    "        raise ValueError(\"Correlation and p-value should be scalars, not matrices.\")\n",
    "\n",
    "    return corr, p_value, method\n",
    "\n",
    "# Step 4: Adjust p-values using Benjamini-Hochberg FDR\n",
    "def adjust_p_values(p_values):\n",
    "    _, p_adj_all, _, _ = multipletests(p_values, method='fdr_bh')\n",
    "    return p_adj_all\n",
    "\n",
    "# Main function to calculate correlations and filter significant results\n",
    "def calculate_correlation(taxonomy_data, pathway_data, selected_genus, up_path, down_path, label=\"Pathway\", alpha=0.05):\n",
    "    taxonomy_selected = filter_data(taxonomy_data, selected_genus)\n",
    "    up_path_data = filter_data(pathway_data, up_path)\n",
    "    down_path_data = filter_data(pathway_data, down_path)\n",
    "\n",
    "    results = []\n",
    "\n",
    "    # Step 5: Run the correlation computation for each pair\n",
    "    for tax_col in taxonomy_selected.columns:\n",
    "        for path_col in up_path_data.columns:\n",
    "            corr, p_value, method = apply_correlation(taxonomy_selected[tax_col], up_path_data[path_col], alpha)\n",
    "            results.append((tax_col, path_col, corr, p_value, method, 'Upregulated'))\n",
    "        \n",
    "        for path_col in down_path_data.columns:\n",
    "            corr, p_value, method = apply_correlation(taxonomy_selected[tax_col], down_path_data[path_col], alpha)\n",
    "            results.append((tax_col, path_col, corr, p_value, method, 'Downregulated'))\n",
    "\n",
    "    # Step 6: Adjust p-values and filter significant correlations\n",
    "    p_values = [result[3] for result in results]\n",
    "    p_adj = adjust_p_values(p_values)\n",
    "\n",
    "    # Add adjusted p-values to results without modifying values with the label\n",
    "    final_results = []\n",
    "    for i, (tax_col, path_col, corr, p_value, method, reg_type) in enumerate(results):\n",
    "        final_results.append((tax_col, path_col, corr, p_value, p_adj[i], method, reg_type))\n",
    "\n",
    "    # Convert to DataFrame with dynamic column names\n",
    "    column_names = [\"Taxonomy\", label, f\"{label}_Correlation\", \n",
    "                    f\"{label}_p_value\", f\"{label}_p_value_adjusted\", \n",
    "                    f\"{label}_Correlation_Method\", f\"{label}_Regulation_Type\"]\n",
    "    \n",
    "    correlation_df = pd.DataFrame(final_results, columns=column_names)\n",
    "\n",
    "    # Filter for significant correlations\n",
    "    significant_corr_df = correlation_df[(correlation_df[f'{label}_p_value_adjusted'] < alpha) & \n",
    "                                         (abs(correlation_df[f'{label}_Correlation']) > 0.2)]\n",
    "\n",
    "    return significant_corr_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Significant Correlations:\n",
      "          Taxonomy  Pathway  Pathway_Correlation  Pathway_p_value  \\\n",
      "0    Desulfovibrio  ko04024             0.238176     1.451721e-07   \n",
      "3    Desulfovibrio  ko04530             0.237698     1.539914e-07   \n",
      "4    Desulfovibrio  ko04151             0.228665     4.589294e-07   \n",
      "5    Desulfovibrio  ko04141             0.213168     2.697765e-06   \n",
      "6    Desulfovibrio  ko01053             0.200167     1.081236e-05   \n",
      "..             ...      ...                  ...              ...   \n",
      "151      Ralstonia  ko00220             0.230607     3.642803e-07   \n",
      "154      Ralstonia  ko05231            -0.208087     4.689993e-06   \n",
      "155      Ralstonia  ko00430            -0.214795     2.253267e-06   \n",
      "160      Ralstonia  ko00471            -0.212862     2.790216e-06   \n",
      "162      Ralstonia  ko00830            -0.226409     5.987259e-07   \n",
      "\n",
      "     Pathway_p_value_adjusted Pathway_Correlation_Method  \\\n",
      "0                3.933695e-07                   Spearman   \n",
      "3                4.106437e-07                   Spearman   \n",
      "4                1.117393e-06                   Spearman   \n",
      "5                5.810570e-06                   Spearman   \n",
      "6                2.064179e-05                   Spearman   \n",
      "..                        ...                        ...   \n",
      "151              8.999867e-07                   Spearman   \n",
      "154              9.608765e-06                   Spearman   \n",
      "155              4.916220e-06                   Spearman   \n",
      "160              5.933623e-06                   Spearman   \n",
      "162              1.397027e-06                   Spearman   \n",
      "\n",
      "    Pathway_Regulation_Type  \n",
      "0               Upregulated  \n",
      "3               Upregulated  \n",
      "4               Upregulated  \n",
      "5               Upregulated  \n",
      "6               Upregulated  \n",
      "..                      ...  \n",
      "151             Upregulated  \n",
      "154           Downregulated  \n",
      "155           Downregulated  \n",
      "160           Downregulated  \n",
      "162           Downregulated  \n",
      "\n",
      "[88 rows x 7 columns]\n",
      "       Taxonomy  Pathway  Pathway_Correlation  Pathway_p_value  \\\n",
      "2   Pseudomonas  ko00601            -0.220704     1.158474e-06   \n",
      "9   Pseudomonas  ko00472             0.237023     1.673411e-07   \n",
      "13  Pseudomonas  ko04150             0.276996     7.838723e-10   \n",
      "14  Pseudomonas  ko03320             0.226752     5.750663e-07   \n",
      "23  Pseudomonas  ko04975             0.231819     3.150122e-07   \n",
      "36  Simonsiella  ko00984             0.423853     3.549182e-22   \n",
      "37  Simonsiella  ko04150             0.244574     6.508678e-08   \n",
      "38  Simonsiella  ko03320             0.290502     1.038739e-10   \n",
      "41  Simonsiella  ko04750             0.428104     1.231596e-22   \n",
      "43  Simonsiella  ko00944             0.443348     2.441962e-24   \n",
      "44  Simonsiella  ko00404             0.235679     1.973172e-07   \n",
      "45  Simonsiella  ko04330             0.412440     5.654703e-21   \n",
      "\n",
      "    Pathway_p_value_adjusted Pathway_Correlation_Method  \\\n",
      "2               4.633897e-06                   Spearman   \n",
      "9               1.004046e-06                   Spearman   \n",
      "13              6.270978e-09                   Spearman   \n",
      "14              2.509380e-06                   Spearman   \n",
      "23              1.512059e-06                   Spearman   \n",
      "36              5.678691e-21                   Spearman   \n",
      "37              4.463093e-07                   Spearman   \n",
      "38              9.971899e-10                   Spearman   \n",
      "41              2.955829e-21                   Spearman   \n",
      "43              1.172142e-22                   Spearman   \n",
      "44              1.052358e-06                   Spearman   \n",
      "45              6.785643e-20                   Spearman   \n",
      "\n",
      "   Pathway_Regulation_Type  \n",
      "2              Upregulated  \n",
      "9            Downregulated  \n",
      "13           Downregulated  \n",
      "14           Downregulated  \n",
      "23           Downregulated  \n",
      "36           Downregulated  \n",
      "37           Downregulated  \n",
      "38           Downregulated  \n",
      "41           Downregulated  \n",
      "43           Downregulated  \n",
      "44           Downregulated  \n",
      "45           Downregulated  \n"
     ]
    }
   ],
   "source": [
    "up_regulated_genus = calculate_correlation(tax_data, path_data,up_genus, up_path, down_path)\n",
    "down_regulated_genus = calculate_correlation(tax_data, path_data, down_genus, up_path, down_path)\n",
    "# Display the significant correlations\n",
    "print(\"Significant Correlations:\")\n",
    "print(up_regulated_genus)\n",
    "print(down_regulated_genus)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>KEGG_no</th>\n",
       "      <th>pathway_kegg_no</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ko00010</td>\n",
       "      <td>Glycolysis / Gluconeogenesis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ko00020</td>\n",
       "      <td>Citrate cycle (TCA cycle)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ko00030</td>\n",
       "      <td>Pentose phosphate pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ko00040</td>\n",
       "      <td>Pentose and glucuronate interconversions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ko00051</td>\n",
       "      <td>Fructose and mannose metabolism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>441</th>\n",
       "      <td>ko05410</td>\n",
       "      <td>Hypertrophic cardiomyopathy (HCM)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>442</th>\n",
       "      <td>ko05412</td>\n",
       "      <td>Arrhythmogenic right ventricular cardiomyopath...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>443</th>\n",
       "      <td>ko05414</td>\n",
       "      <td>Dilated cardiomyopathy (DCM)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>444</th>\n",
       "      <td>ko05416</td>\n",
       "      <td>Viral myocarditis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>445</th>\n",
       "      <td>ko05418</td>\n",
       "      <td>Fluid shear stress and atherosclerosis</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>446 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     KEGG_no                                    pathway_kegg_no\n",
       "0    ko00010                       Glycolysis / Gluconeogenesis\n",
       "1    ko00020                          Citrate cycle (TCA cycle)\n",
       "2    ko00030                          Pentose phosphate pathway\n",
       "3    ko00040           Pentose and glucuronate interconversions\n",
       "4    ko00051                    Fructose and mannose metabolism\n",
       "..       ...                                                ...\n",
       "441  ko05410                  Hypertrophic cardiomyopathy (HCM)\n",
       "442  ko05412  Arrhythmogenic right ventricular cardiomyopath...\n",
       "443  ko05414                       Dilated cardiomyopathy (DCM)\n",
       "444  ko05416                                  Viral myocarditis\n",
       "445  ko05418             Fluid shear stress and atherosclerosis\n",
       "\n",
       "[446 rows x 2 columns]"
      ]
     },
     "execution_count": 275,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "pathway_label = pd.read_csv(path+\"pathway_label_list.csv\")\n",
    "pathway_label"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Assuming you have two DataFrames:\n",
    "# 1. `correlation_df`: The correlation results DataFrame (your provided correlation results)\n",
    "# 2. `kegg_df`: The KEGG pathway dataset with `KEGG_no` and `pathway_kegg_no`\n",
    "\n",
    "# Example DataFrames (replace with your actual data)\n",
    "# correlation_df = pd.read_csv('correlation_results.csv')  # Load your correlation results if from CSV\n",
    "\n",
    "# Perform the merge operation to add pathway labels to your correlation results\n",
    "up_merged_df = pd.merge(up_regulated_genus, pathway_label, left_on='Pathway', right_on='KEGG_no', how='left')\n",
    "\n",
    "# Drop the now redundant 'KEGG_no' column\n",
    "up_merged_df.drop(columns=['KEGG_no'], inplace=True)\n",
    "\n",
    "# Optionally rearrange columns for readability\n",
    "up_merged_df = up_merged_df[[\"Taxonomy\", \"Pathway\", \"pathway_kegg_no\",\"Pathway_Correlation\", \"Pathway_p_value\", \"Pathway_p_value_adjusted\", \"Pathway_Correlation_Method\", \"Pathway_Regulation_Type\"]]\n",
    "\n",
    "down_merged_df = pd.merge(down_regulated_genus, pathway_label, left_on='Pathway', right_on='KEGG_no', how='left')\n",
    "\n",
    "# Drop the now redundant 'KEGG_no' column\n",
    "down_merged_df.drop(columns=['KEGG_no'], inplace=True)\n",
    "down_merged_df = down_merged_df[[\"Taxonomy\", \"Pathway\", \"pathway_kegg_no\",\"Pathway_Correlation\", \"Pathway_p_value\", \"Pathway_p_value_adjusted\", \"Pathway_Correlation_Method\", \"Pathway_Regulation_Type\"]]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['D-Arginine and D-ornithine metabolism',\n",
       "       'Choline metabolism in cancer',\n",
       "       'Taurine and hypotaurine metabolism', 'Prodigiosin biosynthesis',\n",
       "       'D-Glutamine and D-glutamate metabolism', 'Retinol metabolism',\n",
       "       'Ether lipid metabolism', 'Steroid degradation',\n",
       "       'mTOR signaling pathway',\n",
       "       'Inflammatory mediator regulation of TRP channels',\n",
       "       'Flavone and flavonol biosynthesis', 'Notch signaling pathway',\n",
       "       'PPAR signaling pathway', 'Staurosporine biosynthesis',\n",
       "       'Fat digestion and absorption'], dtype=object)"
      ]
     },
     "execution_count": 277,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "up_path_name = up_merged_df[up_merged_df[\"Pathway\"].isin(up_path)]\n",
    "up_path_name.pathway_kegg_no.unique()\n",
    "down_path_name = up_merged_df[up_merged_df[\"Pathway\"].isin(down_path)]\n",
    "down_path_name.pathway_kegg_no.unique()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 기존 선정되었던 orthology 선정"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "selected_ortho_data = pd.read_excel(output_path+'/orthology_analysis_results.xlsx',index_col = 0)\n",
    "selected_ortho_list = selected_ortho_data['Orthology_kegg_no']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 그 중 모든 값이 0인 orthology 제거"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>group_1</th>\n",
       "      <th>group_2</th>\n",
       "      <th>K00001_x</th>\n",
       "      <th>K00001_y</th>\n",
       "      <th>K00029</th>\n",
       "      <th>K00059</th>\n",
       "      <th>K00061</th>\n",
       "      <th>K00121</th>\n",
       "      <th>K00145</th>\n",
       "      <th>K00157</th>\n",
       "      <th>...</th>\n",
       "      <th>K21791</th>\n",
       "      <th>K21792</th>\n",
       "      <th>K22476</th>\n",
       "      <th>K22477</th>\n",
       "      <th>K22478</th>\n",
       "      <th>K08267</th>\n",
       "      <th>K20403</th>\n",
       "      <th>K18435</th>\n",
       "      <th>K09554</th>\n",
       "      <th>K22387</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>study_no</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>H1700001</th>\n",
       "      <td>Cancer</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000152</td>\n",
       "      <td>0.000152</td>\n",
       "      <td>0.000256</td>\n",
       "      <td>0.001308</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>1.880000e-06</td>\n",
       "      <td>...</td>\n",
       "      <td>5.960000e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.090000e-06</td>\n",
       "      <td>0.000086</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1700002</th>\n",
       "      <td>Cancer</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000255</td>\n",
       "      <td>0.000255</td>\n",
       "      <td>0.000385</td>\n",
       "      <td>0.001398</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000050</td>\n",
       "      <td>0.000205</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>1.800000e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1700003</th>\n",
       "      <td>Cancer</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000121</td>\n",
       "      <td>0.000121</td>\n",
       "      <td>0.000641</td>\n",
       "      <td>0.001466</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000226</td>\n",
       "      <td>0.000250</td>\n",
       "      <td>1.100000e-07</td>\n",
       "      <td>...</td>\n",
       "      <td>2.000000e-08</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.600000e-07</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1700004</th>\n",
       "      <td>Cancer</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000242</td>\n",
       "      <td>0.001331</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0.000139</td>\n",
       "      <td>5.300000e-07</td>\n",
       "      <td>...</td>\n",
       "      <td>3.140000e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.100000e-06</td>\n",
       "      <td>0.000036</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1700005</th>\n",
       "      <td>Cancer</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000143</td>\n",
       "      <td>0.000143</td>\n",
       "      <td>0.000473</td>\n",
       "      <td>0.001274</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.000140</td>\n",
       "      <td>2.300000e-07</td>\n",
       "      <td>...</td>\n",
       "      <td>2.500000e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.300000e-07</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1900931</th>\n",
       "      <td>Control</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000076</td>\n",
       "      <td>0.000076</td>\n",
       "      <td>0.000641</td>\n",
       "      <td>0.001260</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000109</td>\n",
       "      <td>0.000148</td>\n",
       "      <td>6.000000e-08</td>\n",
       "      <td>...</td>\n",
       "      <td>3.000000e-08</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.200000e-07</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1900932</th>\n",
       "      <td>Control</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000051</td>\n",
       "      <td>0.000051</td>\n",
       "      <td>0.000733</td>\n",
       "      <td>0.001354</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000194</td>\n",
       "      <td>0.000163</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>2.100000e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.200000e-06</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1900933</th>\n",
       "      <td>Control</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000126</td>\n",
       "      <td>0.000126</td>\n",
       "      <td>0.000593</td>\n",
       "      <td>0.001266</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000104</td>\n",
       "      <td>0.000192</td>\n",
       "      <td>2.200000e-07</td>\n",
       "      <td>...</td>\n",
       "      <td>1.900000e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.110000e-06</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1900936</th>\n",
       "      <td>Control</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000364</td>\n",
       "      <td>0.000364</td>\n",
       "      <td>0.000281</td>\n",
       "      <td>0.001421</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>0.000357</td>\n",
       "      <td>2.600000e-07</td>\n",
       "      <td>...</td>\n",
       "      <td>6.000000e-08</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.400000e-07</td>\n",
       "      <td>0.000019</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1900939</th>\n",
       "      <td>Control</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000324</td>\n",
       "      <td>0.000324</td>\n",
       "      <td>0.000316</td>\n",
       "      <td>0.001428</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000060</td>\n",
       "      <td>0.000319</td>\n",
       "      <td>3.300000e-07</td>\n",
       "      <td>...</td>\n",
       "      <td>8.000000e-08</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.300000e-07</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>476 rows × 792 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          group_1  group_2  K00001_x  K00001_y    K00029    K00059  K00061  \\\n",
       "study_no                                                                     \n",
       "H1700001   Cancer        1  0.000152  0.000152  0.000256  0.001308     0.0   \n",
       "H1700002   Cancer        1  0.000255  0.000255  0.000385  0.001398     0.0   \n",
       "H1700003   Cancer        1  0.000121  0.000121  0.000641  0.001466     0.0   \n",
       "H1700004   Cancer        1  0.000220  0.000220  0.000242  0.001331     0.0   \n",
       "H1700005   Cancer        1  0.000143  0.000143  0.000473  0.001274     0.0   \n",
       "...           ...      ...       ...       ...       ...       ...     ...   \n",
       "H1900931  Control        0  0.000076  0.000076  0.000641  0.001260     0.0   \n",
       "H1900932  Control        0  0.000051  0.000051  0.000733  0.001354     0.0   \n",
       "H1900933  Control        0  0.000126  0.000126  0.000593  0.001266     0.0   \n",
       "H1900936  Control        0  0.000364  0.000364  0.000281  0.001421     0.0   \n",
       "H1900939  Control        0  0.000324  0.000324  0.000316  0.001428     0.0   \n",
       "\n",
       "            K00121    K00145        K00157  ...        K21791  K21792  \\\n",
       "study_no                                    ...                         \n",
       "H1700001  0.000029  0.000098  1.880000e-06  ...  5.960000e-06     0.0   \n",
       "H1700002  0.000050  0.000205  0.000000e+00  ...  1.800000e-07     0.0   \n",
       "H1700003  0.000226  0.000250  1.100000e-07  ...  2.000000e-08     0.0   \n",
       "H1700004  0.000010  0.000139  5.300000e-07  ...  3.140000e-06     0.0   \n",
       "H1700005  0.000017  0.000140  2.300000e-07  ...  2.500000e-07     0.0   \n",
       "...            ...       ...           ...  ...           ...     ...   \n",
       "H1900931  0.000109  0.000148  6.000000e-08  ...  3.000000e-08     0.0   \n",
       "H1900932  0.000194  0.000163  0.000000e+00  ...  2.100000e-07     0.0   \n",
       "H1900933  0.000104  0.000192  2.200000e-07  ...  1.900000e-07     0.0   \n",
       "H1900936  0.000054  0.000357  2.600000e-07  ...  6.000000e-08     0.0   \n",
       "H1900939  0.000060  0.000319  3.300000e-07  ...  8.000000e-08     0.0   \n",
       "\n",
       "                K22476    K22477  K22478  K08267  K20403  K18435  K09554  \\\n",
       "study_no                                                                   \n",
       "H1700001  1.090000e-06  0.000086     0.0     0.0     0.0     0.0     0.0   \n",
       "H1700002  0.000000e+00  0.000010     0.0     0.0     0.0     0.0     0.0   \n",
       "H1700003  3.600000e-07  0.000009     0.0     0.0     0.0     0.0     0.0   \n",
       "H1700004  1.100000e-06  0.000036     0.0     0.0     0.0     0.0     0.0   \n",
       "H1700005  3.300000e-07  0.000009     0.0     0.0     0.0     0.0     0.0   \n",
       "...                ...       ...     ...     ...     ...     ...     ...   \n",
       "H1900931  7.200000e-07  0.000014     0.0     0.0     0.0     0.0     0.0   \n",
       "H1900932  1.200000e-06  0.000013     0.0     0.0     0.0     0.0     0.0   \n",
       "H1900933  1.110000e-06  0.000025     0.0     0.0     0.0     0.0     0.0   \n",
       "H1900936  7.400000e-07  0.000019     0.0     0.0     0.0     0.0     0.0   \n",
       "H1900939  7.300000e-07  0.000022     0.0     0.0     0.0     0.0     0.0   \n",
       "\n",
       "          K22387  \n",
       "study_no          \n",
       "H1700001     0.0  \n",
       "H1700002     0.0  \n",
       "H1700003     0.0  \n",
       "H1700004     0.0  \n",
       "H1700005     0.0  \n",
       "...          ...  \n",
       "H1900931     0.0  \n",
       "H1900932     0.0  \n",
       "H1900933     0.0  \n",
       "H1900936     0.0  \n",
       "H1900939     0.0  \n",
       "\n",
       "[476 rows x 792 columns]"
      ]
     },
     "execution_count": 279,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sod = ortho_data[selected_ortho_list]\n",
    "sod = sod.loc[:, sod.sum() != 0]\n",
    "label = ortho_data.iloc[:,:3]\n",
    "# Merge based on the \"index\" column\n",
    "otho_df = pd.merge(label, sod, how='left', left_index=True, right_index=True)\n",
    "otho_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 위에서 확인된 789개의 orthology와 genus간의 Correlation analysis 진행"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "selected_ortho_data.dropna().reset_index(drop=True,inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "up_regulated_ortho = selected_ortho_data[(selected_ortho_data['FC_value'] >= 2) & (selected_ortho_data['Wilcoxon_p'] <= 0.05) & (selected_ortho_data['LogReg_p_fdr'] <= 0.05)]\n",
    "down_regulated_ortho = selected_ortho_data[(selected_ortho_data['FC_value'] <= 0.5) & (selected_ortho_data['Wilcoxon_p'] <= 0.05) & (selected_ortho_data['LogReg_p_fdr'] <= 0.05)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "up_regulated_ortho.to_excel(output_path + \"/up_regulated_ortho.xlsx\")\n",
    "down_regulated_ortho.to_excel(output_path+ \"/down_regulated_ortho.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "down_ortho = down_regulated_ortho.Orthology_kegg_no.to_list()\n",
    "up_ortho = up_regulated_ortho.Orthology_kegg_no.unique().tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Pathway_ID</th>\n",
       "      <th>Orthology_kegg_no</th>\n",
       "      <th>Orthology_Name</th>\n",
       "      <th>Control_mean</th>\n",
       "      <th>Cancer_mean</th>\n",
       "      <th>FC_value</th>\n",
       "      <th>Wilcoxon_p</th>\n",
       "      <th>log2FC_value</th>\n",
       "      <th>LogReg_p_univ</th>\n",
       "      <th>LogReg_p_fdr</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>ko00430</td>\n",
       "      <td>K03119</td>\n",
       "      <td>taurine dioxygenase</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>6.757609e-07</td>\n",
       "      <td>0.283400</td>\n",
       "      <td>3.287933e-02</td>\n",
       "      <td>-1.819089</td>\n",
       "      <td>4.375793e-20</td>\n",
       "      <td>6.127878e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>453</th>\n",
       "      <td>ko00601</td>\n",
       "      <td>K07966</td>\n",
       "      <td>beta-1,4-galactosyltransferase 1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>1.778478e-06</td>\n",
       "      <td>0.369554</td>\n",
       "      <td>5.579062e-16</td>\n",
       "      <td>-1.436143</td>\n",
       "      <td>1.135174e-15</td>\n",
       "      <td>3.265155e-14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522</th>\n",
       "      <td>ko04141</td>\n",
       "      <td>K09523</td>\n",
       "      <td>DnaJ homolog subfamily C member 3</td>\n",
       "      <td>0.000031</td>\n",
       "      <td>1.475500e-05</td>\n",
       "      <td>0.475648</td>\n",
       "      <td>9.928891e-13</td>\n",
       "      <td>-1.072034</td>\n",
       "      <td>3.316219e-06</td>\n",
       "      <td>2.053419e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>581</th>\n",
       "      <td>ko04120</td>\n",
       "      <td>K10610</td>\n",
       "      <td>DNA damage-binding protein 1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>1.563478e-06</td>\n",
       "      <td>0.312862</td>\n",
       "      <td>2.532964e-09</td>\n",
       "      <td>-1.676402</td>\n",
       "      <td>1.087053e-07</td>\n",
       "      <td>8.226478e-07</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Pathway_ID Orthology_kegg_no                     Orthology_Name  \\\n",
       "149    ko00430            K03119                taurine dioxygenase   \n",
       "453    ko00601            K07966   beta-1,4-galactosyltransferase 1   \n",
       "522    ko04141            K09523  DnaJ homolog subfamily C member 3   \n",
       "581    ko04120            K10610       DNA damage-binding protein 1   \n",
       "\n",
       "     Control_mean   Cancer_mean  FC_value    Wilcoxon_p  log2FC_value  \\\n",
       "149      0.000002  6.757609e-07  0.283400  3.287933e-02     -1.819089   \n",
       "453      0.000005  1.778478e-06  0.369554  5.579062e-16     -1.436143   \n",
       "522      0.000031  1.475500e-05  0.475648  9.928891e-13     -1.072034   \n",
       "581      0.000005  1.563478e-06  0.312862  2.532964e-09     -1.676402   \n",
       "\n",
       "     LogReg_p_univ  LogReg_p_fdr  \n",
       "149   4.375793e-20  6.127878e-18  \n",
       "453   1.135174e-15  3.265155e-14  \n",
       "522   3.316219e-06  2.053419e-05  \n",
       "581   1.087053e-07  8.226478e-07  "
      ]
     },
     "execution_count": 284,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "down_regulated_ortho"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "up_tax_ortho_correlation = calculate_correlation(tax_data, ortho_data,up_genus, up_ortho, down_ortho,label='Orthology')\n",
    "down_tax_ortho_correlation = calculate_correlation(tax_data, ortho_data, down_genus, up_ortho, down_ortho, label=\"Orthology\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "up_tax_ortho_correlation.to_excel(output_path+\"/Correlation_analysis_up_genus_orthology.xlsx\",index=False)\n",
    "down_tax_ortho_correlation.to_excel(output_path+\"/Correlation_analysis_down_genus_orthology.xlsx\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ortholog</th>\n",
       "      <th>Definition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>K12369</td>\n",
       "      <td>dipeptide transport system permease protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>K02389</td>\n",
       "      <td>flagellar basal-body rod modification protein ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>K00839</td>\n",
       "      <td>(S)-ureidoglycine---glyoxylate transaminase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>K15345</td>\n",
       "      <td>secreted effector protein SseC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>K06977</td>\n",
       "      <td>uncharacterized protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12494</th>\n",
       "      <td>K15744</td>\n",
       "      <td>zeta-carotene isomerase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12495</th>\n",
       "      <td>K12705</td>\n",
       "      <td>8-demethylnovobiocic acid C8-methyltransferase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12496</th>\n",
       "      <td>K11534</td>\n",
       "      <td>DeoR family transcriptional regulator, deoxyri...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12497</th>\n",
       "      <td>K17262</td>\n",
       "      <td>tubulin-specific chaperone B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12498</th>\n",
       "      <td>K15658</td>\n",
       "      <td>arthrofactin-type cyclic lipopeptide synthetase A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12499 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Ortholog                                         Definition\n",
       "0       K12369        dipeptide transport system permease protein\n",
       "1       K02389  flagellar basal-body rod modification protein ...\n",
       "2       K00839        (S)-ureidoglycine---glyoxylate transaminase\n",
       "3       K15345                     secreted effector protein SseC\n",
       "4       K06977                            uncharacterized protein\n",
       "...        ...                                                ...\n",
       "12494   K15744                            zeta-carotene isomerase\n",
       "12495   K12705     8-demethylnovobiocic acid C8-methyltransferase\n",
       "12496   K11534  DeoR family transcriptional regulator, deoxyri...\n",
       "12497   K17262                       tubulin-specific chaperone B\n",
       "12498   K15658  arthrofactin-type cyclic lipopeptide synthetase A\n",
       "\n",
       "[12499 rows x 2 columns]"
      ]
     },
     "execution_count": 287,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ortho_label"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge and drop 'Ortholog' in one step\n",
    "up_tax_ortho_correlation = (\n",
    "    pd.merge(up_tax_ortho_correlation, ortho_label, how='left', left_on='Orthology', right_on='Ortholog')\n",
    "    .drop('Ortholog', axis=1)\n",
    ")\n",
    "down_tax_ortho_correlation = (\n",
    "    pd.merge(down_tax_ortho_correlation, ortho_label, how = 'left', left_on = 'Orthology', right_on = 'Ortholog').drop(\"Ortholog\",axis=1)\n",
    ")\n",
    "\n",
    "# Rearrange the columns\n",
    "cols = up_tax_ortho_correlation.columns\n",
    "up_tax_ortho_correlation = up_tax_ortho_correlation[[cols[0], cols[1], cols[-1]] + list(cols[2:-1])]\n",
    "down_tax_ortho_correlation = down_tax_ortho_correlation[[cols[0], cols[1], cols[-1]] + list(cols[2:-1])]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mapping_df = pd.read_excel(path+'/pathway_Orthology_map.xlsx',sheet_name='Sheet2')\n",
    "# Remove all kinds of whitespace (including \\xa0) from 'Orthology_kegg_no' column in mapping_df\n",
    "mapping_df['Orthology_kegg_no'] = mapping_df['Orthology_kegg_no'].str.replace(r'\\s+', '', regex=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 290,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'K00485' in mapping_df['Orthology_kegg_no'].values\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tmp1 = pd.merge(\n",
    "    up_tax_ortho_correlation,\n",
    "    mapping_df[['Pathway_ID', 'Orthology_kegg_no']],\n",
    "    how='left',\n",
    "    left_on=\"Orthology\",\n",
    "    right_on=\"Orthology_kegg_no\"\n",
    ")\n",
    "tmp2 = pd.merge(\n",
    "    down_tax_ortho_correlation,\n",
    "    mapping_df[['Pathway_ID', 'Orthology_kegg_no']],\n",
    "    how='left',\n",
    "    left_on=\"Orthology\",\n",
    "    right_on=\"Orthology_kegg_no\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Correct way to merge on multiple columns\n",
    "up_results = pd.merge(\n",
    "    up_merged_df,\n",
    "    tmp1,\n",
    "    how='right',\n",
    "    left_on=['Taxonomy', 'Pathway'],\n",
    "    right_on=['Taxonomy', 'Pathway_ID']\n",
    ")\n",
    "down_results = pd.merge(\n",
    "    down_merged_df,\n",
    "    tmp2,\n",
    "    how='right',\n",
    "    left_on=['Taxonomy', 'Pathway'],\n",
    "    right_on=['Taxonomy', 'Pathway_ID']\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "up_results.to_excel(output_path+\"/up_results.xlsx\")\n",
    "down_results.to_excel(output_path+\"/down_results.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Taxonomy</th>\n",
       "      <th>Pathway</th>\n",
       "      <th>pathway_kegg_no</th>\n",
       "      <th>Pathway_Correlation</th>\n",
       "      <th>Pathway_p_value</th>\n",
       "      <th>Pathway_p_value_adjusted</th>\n",
       "      <th>Pathway_Correlation_Method</th>\n",
       "      <th>Pathway_Regulation_Type</th>\n",
       "      <th>Orthology</th>\n",
       "      <th>Definition</th>\n",
       "      <th>Orthology_Correlation</th>\n",
       "      <th>Orthology_p_value</th>\n",
       "      <th>Orthology_p_value_adjusted</th>\n",
       "      <th>Orthology_Correlation_Method</th>\n",
       "      <th>Orthology_Regulation_Type</th>\n",
       "      <th>Pathway_ID</th>\n",
       "      <th>Orthology_kegg_no</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Desulfovibrio</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K00485</td>\n",
       "      <td>dimethylaniline monooxygenase (N-oxide forming)</td>\n",
       "      <td>0.244840</td>\n",
       "      <td>6.291979e-08</td>\n",
       "      <td>2.202193e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko00430</td>\n",
       "      <td>K00485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Desulfovibrio</td>\n",
       "      <td>ko00220</td>\n",
       "      <td>Arginine biosynthesis</td>\n",
       "      <td>0.206622</td>\n",
       "      <td>5.486889e-06</td>\n",
       "      <td>1.084467e-05</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K01476</td>\n",
       "      <td>arginase</td>\n",
       "      <td>0.221214</td>\n",
       "      <td>1.092854e-06</td>\n",
       "      <td>3.059992e-06</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko00220</td>\n",
       "      <td>K01476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Desulfovibrio</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K03371</td>\n",
       "      <td>alpha-N-acetyl-neuraminate alpha-2,8-sialyltra...</td>\n",
       "      <td>0.311932</td>\n",
       "      <td>3.351986e-12</td>\n",
       "      <td>1.932321e-11</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko00601</td>\n",
       "      <td>K03371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Desulfovibrio</td>\n",
       "      <td>ko04024</td>\n",
       "      <td>cAMP signaling pathway</td>\n",
       "      <td>0.238176</td>\n",
       "      <td>1.451721e-07</td>\n",
       "      <td>3.933695e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K05208</td>\n",
       "      <td>glutamate receptor ionotropic, NMDA 1</td>\n",
       "      <td>0.284686</td>\n",
       "      <td>2.513500e-10</td>\n",
       "      <td>1.172967e-09</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04024</td>\n",
       "      <td>K05208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Fretibacterium</td>\n",
       "      <td>ko00430</td>\n",
       "      <td>Taurine and hypotaurine metabolism</td>\n",
       "      <td>-0.255149</td>\n",
       "      <td>1.644366e-08</td>\n",
       "      <td>6.005512e-08</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>K00485</td>\n",
       "      <td>dimethylaniline monooxygenase (N-oxide forming)</td>\n",
       "      <td>0.425210</td>\n",
       "      <td>2.536291e-22</td>\n",
       "      <td>2.761739e-21</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko00430</td>\n",
       "      <td>K00485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Fretibacterium</td>\n",
       "      <td>ko00220</td>\n",
       "      <td>Arginine biosynthesis</td>\n",
       "      <td>0.353130</td>\n",
       "      <td>1.999101e-15</td>\n",
       "      <td>3.053172e-14</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K01476</td>\n",
       "      <td>arginase</td>\n",
       "      <td>0.483422</td>\n",
       "      <td>3.016141e-29</td>\n",
       "      <td>4.926364e-28</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko00220</td>\n",
       "      <td>K01476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Fretibacterium</td>\n",
       "      <td>ko00601</td>\n",
       "      <td>Glycosphingolipid biosynthesis - lacto and neo...</td>\n",
       "      <td>0.239896</td>\n",
       "      <td>1.172644e-07</td>\n",
       "      <td>3.339054e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K03371</td>\n",
       "      <td>alpha-N-acetyl-neuraminate alpha-2,8-sialyltra...</td>\n",
       "      <td>0.656058</td>\n",
       "      <td>6.511411e-60</td>\n",
       "      <td>3.190591e-58</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko00601</td>\n",
       "      <td>K03371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Fretibacterium</td>\n",
       "      <td>ko00430</td>\n",
       "      <td>Taurine and hypotaurine metabolism</td>\n",
       "      <td>-0.255149</td>\n",
       "      <td>1.644366e-08</td>\n",
       "      <td>6.005512e-08</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>K03852</td>\n",
       "      <td>sulfoacetaldehyde acetyltransferase</td>\n",
       "      <td>0.294357</td>\n",
       "      <td>5.718457e-11</td>\n",
       "      <td>2.802044e-10</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko00430</td>\n",
       "      <td>K03852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Fretibacterium</td>\n",
       "      <td>ko04024</td>\n",
       "      <td>cAMP signaling pathway</td>\n",
       "      <td>0.403474</td>\n",
       "      <td>4.623776e-20</td>\n",
       "      <td>2.102678e-18</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K05208</td>\n",
       "      <td>glutamate receptor ionotropic, NMDA 1</td>\n",
       "      <td>0.542645</td>\n",
       "      <td>8.438361e-38</td>\n",
       "      <td>2.067398e-36</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04024</td>\n",
       "      <td>K05208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Fretibacterium</td>\n",
       "      <td>ko04151</td>\n",
       "      <td>PI3K-Akt signaling pathway</td>\n",
       "      <td>0.335350</td>\n",
       "      <td>5.649018e-14</td>\n",
       "      <td>7.300269e-13</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K07207</td>\n",
       "      <td>tuberous sclerosis 2</td>\n",
       "      <td>0.248362</td>\n",
       "      <td>4.004823e-08</td>\n",
       "      <td>1.453603e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04151</td>\n",
       "      <td>K07207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Fretibacterium</td>\n",
       "      <td>ko05231</td>\n",
       "      <td>Choline metabolism in cancer</td>\n",
       "      <td>-0.200591</td>\n",
       "      <td>1.034817e-05</td>\n",
       "      <td>1.998267e-05</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>K07207</td>\n",
       "      <td>tuberous sclerosis 2</td>\n",
       "      <td>0.248362</td>\n",
       "      <td>4.004823e-08</td>\n",
       "      <td>1.453603e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko05231</td>\n",
       "      <td>K07207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Fretibacterium</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K07207</td>\n",
       "      <td>tuberous sclerosis 2</td>\n",
       "      <td>0.248362</td>\n",
       "      <td>4.004823e-08</td>\n",
       "      <td>1.453603e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04150</td>\n",
       "      <td>K07207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Fretibacterium</td>\n",
       "      <td>ko04530</td>\n",
       "      <td>Tight junction</td>\n",
       "      <td>0.403130</td>\n",
       "      <td>5.006377e-20</td>\n",
       "      <td>2.102678e-18</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K07298</td>\n",
       "      <td>serine/threonine-protein kinase 11</td>\n",
       "      <td>0.353184</td>\n",
       "      <td>1.978050e-15</td>\n",
       "      <td>1.491145e-14</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04530</td>\n",
       "      <td>K07298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Fretibacterium</td>\n",
       "      <td>ko04151</td>\n",
       "      <td>PI3K-Akt signaling pathway</td>\n",
       "      <td>0.335350</td>\n",
       "      <td>5.649018e-14</td>\n",
       "      <td>7.300269e-13</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K07298</td>\n",
       "      <td>serine/threonine-protein kinase 11</td>\n",
       "      <td>0.353184</td>\n",
       "      <td>1.978050e-15</td>\n",
       "      <td>1.491145e-14</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04151</td>\n",
       "      <td>K07298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Fretibacterium</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K07298</td>\n",
       "      <td>serine/threonine-protein kinase 11</td>\n",
       "      <td>0.353184</td>\n",
       "      <td>1.978050e-15</td>\n",
       "      <td>1.491145e-14</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04150</td>\n",
       "      <td>K07298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Fretibacterium</td>\n",
       "      <td>ko04151</td>\n",
       "      <td>PI3K-Akt signaling pathway</td>\n",
       "      <td>0.335350</td>\n",
       "      <td>5.649018e-14</td>\n",
       "      <td>7.300269e-13</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K07825</td>\n",
       "      <td>guanine nucleotide-binding protein G(I)/G(S)/G...</td>\n",
       "      <td>0.257025</td>\n",
       "      <td>1.279951e-08</td>\n",
       "      <td>4.824430e-08</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04151</td>\n",
       "      <td>K07825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Fretibacterium</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K16438</td>\n",
       "      <td>5-epimerase</td>\n",
       "      <td>0.242690</td>\n",
       "      <td>8.262449e-08</td>\n",
       "      <td>2.792138e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko00404</td>\n",
       "      <td>K16438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Fretibacterium</td>\n",
       "      <td>ko00430</td>\n",
       "      <td>Taurine and hypotaurine metabolism</td>\n",
       "      <td>-0.255149</td>\n",
       "      <td>1.644366e-08</td>\n",
       "      <td>6.005512e-08</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>K03119</td>\n",
       "      <td>taurine dioxygenase</td>\n",
       "      <td>0.236416</td>\n",
       "      <td>1.802854e-07</td>\n",
       "      <td>5.570495e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>ko00430</td>\n",
       "      <td>K03119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Lactobacillus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K00485</td>\n",
       "      <td>dimethylaniline monooxygenase (N-oxide forming)</td>\n",
       "      <td>0.236344</td>\n",
       "      <td>1.818937e-07</td>\n",
       "      <td>5.570495e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko00430</td>\n",
       "      <td>K00485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Lactobacillus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K07207</td>\n",
       "      <td>tuberous sclerosis 2</td>\n",
       "      <td>0.241851</td>\n",
       "      <td>9.181914e-08</td>\n",
       "      <td>2.999425e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04151</td>\n",
       "      <td>K07207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Lactobacillus</td>\n",
       "      <td>ko05231</td>\n",
       "      <td>Choline metabolism in cancer</td>\n",
       "      <td>-0.241708</td>\n",
       "      <td>9.348344e-08</td>\n",
       "      <td>2.707796e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>K07207</td>\n",
       "      <td>tuberous sclerosis 2</td>\n",
       "      <td>0.241851</td>\n",
       "      <td>9.181914e-08</td>\n",
       "      <td>2.999425e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko05231</td>\n",
       "      <td>K07207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Lactobacillus</td>\n",
       "      <td>ko04150</td>\n",
       "      <td>mTOR signaling pathway</td>\n",
       "      <td>-0.220099</td>\n",
       "      <td>1.241190e-06</td>\n",
       "      <td>2.780266e-06</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>K07207</td>\n",
       "      <td>tuberous sclerosis 2</td>\n",
       "      <td>0.241851</td>\n",
       "      <td>9.181914e-08</td>\n",
       "      <td>2.999425e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04150</td>\n",
       "      <td>K07207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Lactobacillus</td>\n",
       "      <td>ko04141</td>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>0.206920</td>\n",
       "      <td>5.315182e-06</td>\n",
       "      <td>1.063036e-05</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K13999</td>\n",
       "      <td>cytoskeleton-associated protein 4</td>\n",
       "      <td>0.317228</td>\n",
       "      <td>1.372855e-12</td>\n",
       "      <td>8.408736e-12</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04141</td>\n",
       "      <td>K13999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Lactobacillus</td>\n",
       "      <td>ko00601</td>\n",
       "      <td>Glycosphingolipid biosynthesis - lacto and neo...</td>\n",
       "      <td>0.242249</td>\n",
       "      <td>8.734495e-08</td>\n",
       "      <td>2.620348e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K07966</td>\n",
       "      <td>beta-1,4-galactosyltransferase 1</td>\n",
       "      <td>-0.351234</td>\n",
       "      <td>2.884086e-15</td>\n",
       "      <td>2.018860e-14</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>ko00601</td>\n",
       "      <td>K07966</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Lactobacillus</td>\n",
       "      <td>ko04141</td>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>0.206920</td>\n",
       "      <td>5.315182e-06</td>\n",
       "      <td>1.063036e-05</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K09523</td>\n",
       "      <td>DnaJ homolog subfamily C member 3</td>\n",
       "      <td>-0.368790</td>\n",
       "      <td>8.800392e-17</td>\n",
       "      <td>7.186986e-16</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>ko04141</td>\n",
       "      <td>K09523</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Leuconostoc</td>\n",
       "      <td>ko00601</td>\n",
       "      <td>Glycosphingolipid biosynthesis - lacto and neo...</td>\n",
       "      <td>0.278198</td>\n",
       "      <td>6.577368e-10</td>\n",
       "      <td>3.348478e-09</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K07966</td>\n",
       "      <td>beta-1,4-galactosyltransferase 1</td>\n",
       "      <td>-0.235259</td>\n",
       "      <td>2.077148e-07</td>\n",
       "      <td>6.168500e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>ko00601</td>\n",
       "      <td>K07966</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Leuconostoc</td>\n",
       "      <td>ko04141</td>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>0.250193</td>\n",
       "      <td>3.158052e-08</td>\n",
       "      <td>1.001043e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K09523</td>\n",
       "      <td>DnaJ homolog subfamily C member 3</td>\n",
       "      <td>-0.213072</td>\n",
       "      <td>2.726491e-06</td>\n",
       "      <td>7.221517e-06</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>ko04141</td>\n",
       "      <td>K09523</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Olsenella</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K00485</td>\n",
       "      <td>dimethylaniline monooxygenase (N-oxide forming)</td>\n",
       "      <td>0.370324</td>\n",
       "      <td>6.419752e-17</td>\n",
       "      <td>5.719416e-16</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko00430</td>\n",
       "      <td>K00485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Olsenella</td>\n",
       "      <td>ko00601</td>\n",
       "      <td>Glycosphingolipid biosynthesis - lacto and neo...</td>\n",
       "      <td>0.258780</td>\n",
       "      <td>1.010653e-08</td>\n",
       "      <td>3.975522e-08</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K03371</td>\n",
       "      <td>alpha-N-acetyl-neuraminate alpha-2,8-sialyltra...</td>\n",
       "      <td>0.343685</td>\n",
       "      <td>1.211167e-14</td>\n",
       "      <td>7.912957e-14</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko00601</td>\n",
       "      <td>K03371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Olsenella</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K13999</td>\n",
       "      <td>cytoskeleton-associated protein 4</td>\n",
       "      <td>0.213570</td>\n",
       "      <td>2.580597e-06</td>\n",
       "      <td>7.024960e-06</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04141</td>\n",
       "      <td>K13999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Parvimonas</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K00485</td>\n",
       "      <td>dimethylaniline monooxygenase (N-oxide forming)</td>\n",
       "      <td>0.722189</td>\n",
       "      <td>6.668970e-78</td>\n",
       "      <td>6.535591e-76</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko00430</td>\n",
       "      <td>K00485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Parvimonas</td>\n",
       "      <td>ko00220</td>\n",
       "      <td>Arginine biosynthesis</td>\n",
       "      <td>0.440959</td>\n",
       "      <td>4.574453e-24</td>\n",
       "      <td>3.842540e-22</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K01476</td>\n",
       "      <td>arginase</td>\n",
       "      <td>0.416132</td>\n",
       "      <td>2.336362e-21</td>\n",
       "      <td>2.289635e-20</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko00220</td>\n",
       "      <td>K01476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Parvimonas</td>\n",
       "      <td>ko00601</td>\n",
       "      <td>Glycosphingolipid biosynthesis - lacto and neo...</td>\n",
       "      <td>0.364617</td>\n",
       "      <td>2.056796e-16</td>\n",
       "      <td>3.455418e-15</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K03371</td>\n",
       "      <td>alpha-N-acetyl-neuraminate alpha-2,8-sialyltra...</td>\n",
       "      <td>0.494270</td>\n",
       "      <td>1.085112e-30</td>\n",
       "      <td>2.126819e-29</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko00601</td>\n",
       "      <td>K03371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Parvimonas</td>\n",
       "      <td>ko04024</td>\n",
       "      <td>cAMP signaling pathway</td>\n",
       "      <td>0.365712</td>\n",
       "      <td>1.648133e-16</td>\n",
       "      <td>3.076515e-15</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K05208</td>\n",
       "      <td>glutamate receptor ionotropic, NMDA 1</td>\n",
       "      <td>0.473591</td>\n",
       "      <td>5.550386e-28</td>\n",
       "      <td>7.770540e-27</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04024</td>\n",
       "      <td>K05208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Parvimonas</td>\n",
       "      <td>ko04151</td>\n",
       "      <td>PI3K-Akt signaling pathway</td>\n",
       "      <td>0.392703</td>\n",
       "      <td>5.314090e-19</td>\n",
       "      <td>1.275382e-17</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K07207</td>\n",
       "      <td>tuberous sclerosis 2</td>\n",
       "      <td>0.234213</td>\n",
       "      <td>2.359218e-07</td>\n",
       "      <td>6.800099e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04151</td>\n",
       "      <td>K07207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Parvimonas</td>\n",
       "      <td>ko05231</td>\n",
       "      <td>Choline metabolism in cancer</td>\n",
       "      <td>-0.233415</td>\n",
       "      <td>2.598821e-07</td>\n",
       "      <td>6.716953e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>K07207</td>\n",
       "      <td>tuberous sclerosis 2</td>\n",
       "      <td>0.234213</td>\n",
       "      <td>2.359218e-07</td>\n",
       "      <td>6.800099e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko05231</td>\n",
       "      <td>K07207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>Parvimonas</td>\n",
       "      <td>ko04150</td>\n",
       "      <td>mTOR signaling pathway</td>\n",
       "      <td>-0.226117</td>\n",
       "      <td>6.195255e-07</td>\n",
       "      <td>1.425757e-06</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>K07207</td>\n",
       "      <td>tuberous sclerosis 2</td>\n",
       "      <td>0.234213</td>\n",
       "      <td>2.359218e-07</td>\n",
       "      <td>6.800099e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04150</td>\n",
       "      <td>K07207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>Parvimonas</td>\n",
       "      <td>ko04530</td>\n",
       "      <td>Tight junction</td>\n",
       "      <td>0.400492</td>\n",
       "      <td>9.173573e-20</td>\n",
       "      <td>3.082321e-18</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K07298</td>\n",
       "      <td>serine/threonine-protein kinase 11</td>\n",
       "      <td>0.278529</td>\n",
       "      <td>6.265783e-10</td>\n",
       "      <td>2.669768e-09</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04530</td>\n",
       "      <td>K07298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Parvimonas</td>\n",
       "      <td>ko04151</td>\n",
       "      <td>PI3K-Akt signaling pathway</td>\n",
       "      <td>0.392703</td>\n",
       "      <td>5.314090e-19</td>\n",
       "      <td>1.275382e-17</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K07298</td>\n",
       "      <td>serine/threonine-protein kinase 11</td>\n",
       "      <td>0.278529</td>\n",
       "      <td>6.265783e-10</td>\n",
       "      <td>2.669768e-09</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04151</td>\n",
       "      <td>K07298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>Parvimonas</td>\n",
       "      <td>ko04150</td>\n",
       "      <td>mTOR signaling pathway</td>\n",
       "      <td>-0.226117</td>\n",
       "      <td>6.195255e-07</td>\n",
       "      <td>1.425757e-06</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>K07298</td>\n",
       "      <td>serine/threonine-protein kinase 11</td>\n",
       "      <td>0.278529</td>\n",
       "      <td>6.265783e-10</td>\n",
       "      <td>2.669768e-09</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04150</td>\n",
       "      <td>K07298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>Parvimonas</td>\n",
       "      <td>ko04151</td>\n",
       "      <td>PI3K-Akt signaling pathway</td>\n",
       "      <td>0.392703</td>\n",
       "      <td>5.314090e-19</td>\n",
       "      <td>1.275382e-17</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K07825</td>\n",
       "      <td>guanine nucleotide-binding protein G(I)/G(S)/G...</td>\n",
       "      <td>0.274484</td>\n",
       "      <td>1.128047e-09</td>\n",
       "      <td>4.606191e-09</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04151</td>\n",
       "      <td>K07825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Ralstonia</td>\n",
       "      <td>ko00220</td>\n",
       "      <td>Arginine biosynthesis</td>\n",
       "      <td>0.230607</td>\n",
       "      <td>3.642803e-07</td>\n",
       "      <td>8.999867e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K01476</td>\n",
       "      <td>arginase</td>\n",
       "      <td>0.308364</td>\n",
       "      <td>6.055013e-12</td>\n",
       "      <td>3.296618e-11</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko00220</td>\n",
       "      <td>K01476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>Ralstonia</td>\n",
       "      <td>ko00430</td>\n",
       "      <td>Taurine and hypotaurine metabolism</td>\n",
       "      <td>-0.214795</td>\n",
       "      <td>2.253267e-06</td>\n",
       "      <td>4.916220e-06</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>K03852</td>\n",
       "      <td>sulfoacetaldehyde acetyltransferase</td>\n",
       "      <td>0.270702</td>\n",
       "      <td>1.937765e-09</td>\n",
       "      <td>7.596038e-09</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko00430</td>\n",
       "      <td>K03852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>Ralstonia</td>\n",
       "      <td>ko04024</td>\n",
       "      <td>cAMP signaling pathway</td>\n",
       "      <td>0.217110</td>\n",
       "      <td>1.740095e-06</td>\n",
       "      <td>3.846526e-06</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>K05208</td>\n",
       "      <td>glutamate receptor ionotropic, NMDA 1</td>\n",
       "      <td>0.280494</td>\n",
       "      <td>4.692875e-10</td>\n",
       "      <td>2.090462e-09</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04024</td>\n",
       "      <td>K05208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>Ralstonia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K07298</td>\n",
       "      <td>serine/threonine-protein kinase 11</td>\n",
       "      <td>0.437778</td>\n",
       "      <td>1.047037e-23</td>\n",
       "      <td>1.282620e-22</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04530</td>\n",
       "      <td>K07298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>Ralstonia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K07298</td>\n",
       "      <td>serine/threonine-protein kinase 11</td>\n",
       "      <td>0.437778</td>\n",
       "      <td>1.047037e-23</td>\n",
       "      <td>1.282620e-22</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04151</td>\n",
       "      <td>K07298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>Ralstonia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K07298</td>\n",
       "      <td>serine/threonine-protein kinase 11</td>\n",
       "      <td>0.437778</td>\n",
       "      <td>1.047037e-23</td>\n",
       "      <td>1.282620e-22</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04150</td>\n",
       "      <td>K07298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>Ralstonia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K16438</td>\n",
       "      <td>5-epimerase</td>\n",
       "      <td>0.304165</td>\n",
       "      <td>1.202252e-11</td>\n",
       "      <td>6.201092e-11</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko00404</td>\n",
       "      <td>K16438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>Ralstonia</td>\n",
       "      <td>ko00430</td>\n",
       "      <td>Taurine and hypotaurine metabolism</td>\n",
       "      <td>-0.214795</td>\n",
       "      <td>2.253267e-06</td>\n",
       "      <td>4.916220e-06</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>K03119</td>\n",
       "      <td>taurine dioxygenase</td>\n",
       "      <td>0.606136</td>\n",
       "      <td>4.436864e-49</td>\n",
       "      <td>1.449376e-47</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>ko00430</td>\n",
       "      <td>K03119</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Taxonomy  Pathway  \\\n",
       "0    Desulfovibrio      NaN   \n",
       "1    Desulfovibrio  ko00220   \n",
       "2    Desulfovibrio      NaN   \n",
       "3    Desulfovibrio  ko04024   \n",
       "4   Fretibacterium  ko00430   \n",
       "5   Fretibacterium  ko00220   \n",
       "6   Fretibacterium  ko00601   \n",
       "7   Fretibacterium  ko00430   \n",
       "8   Fretibacterium  ko04024   \n",
       "9   Fretibacterium  ko04151   \n",
       "10  Fretibacterium  ko05231   \n",
       "11  Fretibacterium      NaN   \n",
       "12  Fretibacterium  ko04530   \n",
       "13  Fretibacterium  ko04151   \n",
       "14  Fretibacterium      NaN   \n",
       "15  Fretibacterium  ko04151   \n",
       "16  Fretibacterium      NaN   \n",
       "17  Fretibacterium  ko00430   \n",
       "18   Lactobacillus      NaN   \n",
       "19   Lactobacillus      NaN   \n",
       "20   Lactobacillus  ko05231   \n",
       "21   Lactobacillus  ko04150   \n",
       "22   Lactobacillus  ko04141   \n",
       "23   Lactobacillus  ko00601   \n",
       "24   Lactobacillus  ko04141   \n",
       "25     Leuconostoc  ko00601   \n",
       "26     Leuconostoc  ko04141   \n",
       "27       Olsenella      NaN   \n",
       "28       Olsenella  ko00601   \n",
       "29       Olsenella      NaN   \n",
       "30      Parvimonas      NaN   \n",
       "31      Parvimonas  ko00220   \n",
       "32      Parvimonas  ko00601   \n",
       "33      Parvimonas  ko04024   \n",
       "34      Parvimonas  ko04151   \n",
       "35      Parvimonas  ko05231   \n",
       "36      Parvimonas  ko04150   \n",
       "37      Parvimonas  ko04530   \n",
       "38      Parvimonas  ko04151   \n",
       "39      Parvimonas  ko04150   \n",
       "40      Parvimonas  ko04151   \n",
       "41       Ralstonia  ko00220   \n",
       "42       Ralstonia  ko00430   \n",
       "43       Ralstonia  ko04024   \n",
       "44       Ralstonia      NaN   \n",
       "45       Ralstonia      NaN   \n",
       "46       Ralstonia      NaN   \n",
       "47       Ralstonia      NaN   \n",
       "48       Ralstonia  ko00430   \n",
       "\n",
       "                                      pathway_kegg_no  Pathway_Correlation  \\\n",
       "0                                                 NaN                  NaN   \n",
       "1                               Arginine biosynthesis             0.206622   \n",
       "2                                                 NaN                  NaN   \n",
       "3                              cAMP signaling pathway             0.238176   \n",
       "4                  Taurine and hypotaurine metabolism            -0.255149   \n",
       "5                               Arginine biosynthesis             0.353130   \n",
       "6   Glycosphingolipid biosynthesis - lacto and neo...             0.239896   \n",
       "7                  Taurine and hypotaurine metabolism            -0.255149   \n",
       "8                              cAMP signaling pathway             0.403474   \n",
       "9                          PI3K-Akt signaling pathway             0.335350   \n",
       "10                       Choline metabolism in cancer            -0.200591   \n",
       "11                                                NaN                  NaN   \n",
       "12                                     Tight junction             0.403130   \n",
       "13                         PI3K-Akt signaling pathway             0.335350   \n",
       "14                                                NaN                  NaN   \n",
       "15                         PI3K-Akt signaling pathway             0.335350   \n",
       "16                                                NaN                  NaN   \n",
       "17                 Taurine and hypotaurine metabolism            -0.255149   \n",
       "18                                                NaN                  NaN   \n",
       "19                                                NaN                  NaN   \n",
       "20                       Choline metabolism in cancer            -0.241708   \n",
       "21                             mTOR signaling pathway            -0.220099   \n",
       "22        Protein processing in endoplasmic reticulum             0.206920   \n",
       "23  Glycosphingolipid biosynthesis - lacto and neo...             0.242249   \n",
       "24        Protein processing in endoplasmic reticulum             0.206920   \n",
       "25  Glycosphingolipid biosynthesis - lacto and neo...             0.278198   \n",
       "26        Protein processing in endoplasmic reticulum             0.250193   \n",
       "27                                                NaN                  NaN   \n",
       "28  Glycosphingolipid biosynthesis - lacto and neo...             0.258780   \n",
       "29                                                NaN                  NaN   \n",
       "30                                                NaN                  NaN   \n",
       "31                              Arginine biosynthesis             0.440959   \n",
       "32  Glycosphingolipid biosynthesis - lacto and neo...             0.364617   \n",
       "33                             cAMP signaling pathway             0.365712   \n",
       "34                         PI3K-Akt signaling pathway             0.392703   \n",
       "35                       Choline metabolism in cancer            -0.233415   \n",
       "36                             mTOR signaling pathway            -0.226117   \n",
       "37                                     Tight junction             0.400492   \n",
       "38                         PI3K-Akt signaling pathway             0.392703   \n",
       "39                             mTOR signaling pathway            -0.226117   \n",
       "40                         PI3K-Akt signaling pathway             0.392703   \n",
       "41                              Arginine biosynthesis             0.230607   \n",
       "42                 Taurine and hypotaurine metabolism            -0.214795   \n",
       "43                             cAMP signaling pathway             0.217110   \n",
       "44                                                NaN                  NaN   \n",
       "45                                                NaN                  NaN   \n",
       "46                                                NaN                  NaN   \n",
       "47                                                NaN                  NaN   \n",
       "48                 Taurine and hypotaurine metabolism            -0.214795   \n",
       "\n",
       "    Pathway_p_value  Pathway_p_value_adjusted Pathway_Correlation_Method  \\\n",
       "0               NaN                       NaN                        NaN   \n",
       "1      5.486889e-06              1.084467e-05                   Spearman   \n",
       "2               NaN                       NaN                        NaN   \n",
       "3      1.451721e-07              3.933695e-07                   Spearman   \n",
       "4      1.644366e-08              6.005512e-08                   Spearman   \n",
       "5      1.999101e-15              3.053172e-14                   Spearman   \n",
       "6      1.172644e-07              3.339054e-07                   Spearman   \n",
       "7      1.644366e-08              6.005512e-08                   Spearman   \n",
       "8      4.623776e-20              2.102678e-18                   Spearman   \n",
       "9      5.649018e-14              7.300269e-13                   Spearman   \n",
       "10     1.034817e-05              1.998267e-05                   Spearman   \n",
       "11              NaN                       NaN                        NaN   \n",
       "12     5.006377e-20              2.102678e-18                   Spearman   \n",
       "13     5.649018e-14              7.300269e-13                   Spearman   \n",
       "14              NaN                       NaN                        NaN   \n",
       "15     5.649018e-14              7.300269e-13                   Spearman   \n",
       "16              NaN                       NaN                        NaN   \n",
       "17     1.644366e-08              6.005512e-08                   Spearman   \n",
       "18              NaN                       NaN                        NaN   \n",
       "19              NaN                       NaN                        NaN   \n",
       "20     9.348344e-08              2.707796e-07                   Spearman   \n",
       "21     1.241190e-06              2.780266e-06                   Spearman   \n",
       "22     5.315182e-06              1.063036e-05                   Spearman   \n",
       "23     8.734495e-08              2.620348e-07                   Spearman   \n",
       "24     5.315182e-06              1.063036e-05                   Spearman   \n",
       "25     6.577368e-10              3.348478e-09                   Spearman   \n",
       "26     3.158052e-08              1.001043e-07                   Spearman   \n",
       "27              NaN                       NaN                        NaN   \n",
       "28     1.010653e-08              3.975522e-08                   Spearman   \n",
       "29              NaN                       NaN                        NaN   \n",
       "30              NaN                       NaN                        NaN   \n",
       "31     4.574453e-24              3.842540e-22                   Spearman   \n",
       "32     2.056796e-16              3.455418e-15                   Spearman   \n",
       "33     1.648133e-16              3.076515e-15                   Spearman   \n",
       "34     5.314090e-19              1.275382e-17                   Spearman   \n",
       "35     2.598821e-07              6.716953e-07                   Spearman   \n",
       "36     6.195255e-07              1.425757e-06                   Spearman   \n",
       "37     9.173573e-20              3.082321e-18                   Spearman   \n",
       "38     5.314090e-19              1.275382e-17                   Spearman   \n",
       "39     6.195255e-07              1.425757e-06                   Spearman   \n",
       "40     5.314090e-19              1.275382e-17                   Spearman   \n",
       "41     3.642803e-07              8.999867e-07                   Spearman   \n",
       "42     2.253267e-06              4.916220e-06                   Spearman   \n",
       "43     1.740095e-06              3.846526e-06                   Spearman   \n",
       "44              NaN                       NaN                        NaN   \n",
       "45              NaN                       NaN                        NaN   \n",
       "46              NaN                       NaN                        NaN   \n",
       "47              NaN                       NaN                        NaN   \n",
       "48     2.253267e-06              4.916220e-06                   Spearman   \n",
       "\n",
       "   Pathway_Regulation_Type Orthology  \\\n",
       "0                      NaN    K00485   \n",
       "1              Upregulated    K01476   \n",
       "2                      NaN    K03371   \n",
       "3              Upregulated    K05208   \n",
       "4            Downregulated    K00485   \n",
       "5              Upregulated    K01476   \n",
       "6              Upregulated    K03371   \n",
       "7            Downregulated    K03852   \n",
       "8              Upregulated    K05208   \n",
       "9              Upregulated    K07207   \n",
       "10           Downregulated    K07207   \n",
       "11                     NaN    K07207   \n",
       "12             Upregulated    K07298   \n",
       "13             Upregulated    K07298   \n",
       "14                     NaN    K07298   \n",
       "15             Upregulated    K07825   \n",
       "16                     NaN    K16438   \n",
       "17           Downregulated    K03119   \n",
       "18                     NaN    K00485   \n",
       "19                     NaN    K07207   \n",
       "20           Downregulated    K07207   \n",
       "21           Downregulated    K07207   \n",
       "22             Upregulated    K13999   \n",
       "23             Upregulated    K07966   \n",
       "24             Upregulated    K09523   \n",
       "25             Upregulated    K07966   \n",
       "26             Upregulated    K09523   \n",
       "27                     NaN    K00485   \n",
       "28             Upregulated    K03371   \n",
       "29                     NaN    K13999   \n",
       "30                     NaN    K00485   \n",
       "31             Upregulated    K01476   \n",
       "32             Upregulated    K03371   \n",
       "33             Upregulated    K05208   \n",
       "34             Upregulated    K07207   \n",
       "35           Downregulated    K07207   \n",
       "36           Downregulated    K07207   \n",
       "37             Upregulated    K07298   \n",
       "38             Upregulated    K07298   \n",
       "39           Downregulated    K07298   \n",
       "40             Upregulated    K07825   \n",
       "41             Upregulated    K01476   \n",
       "42           Downregulated    K03852   \n",
       "43             Upregulated    K05208   \n",
       "44                     NaN    K07298   \n",
       "45                     NaN    K07298   \n",
       "46                     NaN    K07298   \n",
       "47                     NaN    K16438   \n",
       "48           Downregulated    K03119   \n",
       "\n",
       "                                           Definition  Orthology_Correlation  \\\n",
       "0     dimethylaniline monooxygenase (N-oxide forming)               0.244840   \n",
       "1                                            arginase               0.221214   \n",
       "2   alpha-N-acetyl-neuraminate alpha-2,8-sialyltra...               0.311932   \n",
       "3               glutamate receptor ionotropic, NMDA 1               0.284686   \n",
       "4     dimethylaniline monooxygenase (N-oxide forming)               0.425210   \n",
       "5                                            arginase               0.483422   \n",
       "6   alpha-N-acetyl-neuraminate alpha-2,8-sialyltra...               0.656058   \n",
       "7                 sulfoacetaldehyde acetyltransferase               0.294357   \n",
       "8               glutamate receptor ionotropic, NMDA 1               0.542645   \n",
       "9                                tuberous sclerosis 2               0.248362   \n",
       "10                               tuberous sclerosis 2               0.248362   \n",
       "11                               tuberous sclerosis 2               0.248362   \n",
       "12                 serine/threonine-protein kinase 11               0.353184   \n",
       "13                 serine/threonine-protein kinase 11               0.353184   \n",
       "14                 serine/threonine-protein kinase 11               0.353184   \n",
       "15  guanine nucleotide-binding protein G(I)/G(S)/G...               0.257025   \n",
       "16                                        5-epimerase               0.242690   \n",
       "17                                taurine dioxygenase               0.236416   \n",
       "18    dimethylaniline monooxygenase (N-oxide forming)               0.236344   \n",
       "19                               tuberous sclerosis 2               0.241851   \n",
       "20                               tuberous sclerosis 2               0.241851   \n",
       "21                               tuberous sclerosis 2               0.241851   \n",
       "22                  cytoskeleton-associated protein 4               0.317228   \n",
       "23                   beta-1,4-galactosyltransferase 1              -0.351234   \n",
       "24                  DnaJ homolog subfamily C member 3              -0.368790   \n",
       "25                   beta-1,4-galactosyltransferase 1              -0.235259   \n",
       "26                  DnaJ homolog subfamily C member 3              -0.213072   \n",
       "27    dimethylaniline monooxygenase (N-oxide forming)               0.370324   \n",
       "28  alpha-N-acetyl-neuraminate alpha-2,8-sialyltra...               0.343685   \n",
       "29                  cytoskeleton-associated protein 4               0.213570   \n",
       "30    dimethylaniline monooxygenase (N-oxide forming)               0.722189   \n",
       "31                                           arginase               0.416132   \n",
       "32  alpha-N-acetyl-neuraminate alpha-2,8-sialyltra...               0.494270   \n",
       "33              glutamate receptor ionotropic, NMDA 1               0.473591   \n",
       "34                               tuberous sclerosis 2               0.234213   \n",
       "35                               tuberous sclerosis 2               0.234213   \n",
       "36                               tuberous sclerosis 2               0.234213   \n",
       "37                 serine/threonine-protein kinase 11               0.278529   \n",
       "38                 serine/threonine-protein kinase 11               0.278529   \n",
       "39                 serine/threonine-protein kinase 11               0.278529   \n",
       "40  guanine nucleotide-binding protein G(I)/G(S)/G...               0.274484   \n",
       "41                                           arginase               0.308364   \n",
       "42                sulfoacetaldehyde acetyltransferase               0.270702   \n",
       "43              glutamate receptor ionotropic, NMDA 1               0.280494   \n",
       "44                 serine/threonine-protein kinase 11               0.437778   \n",
       "45                 serine/threonine-protein kinase 11               0.437778   \n",
       "46                 serine/threonine-protein kinase 11               0.437778   \n",
       "47                                        5-epimerase               0.304165   \n",
       "48                                taurine dioxygenase               0.606136   \n",
       "\n",
       "    Orthology_p_value  Orthology_p_value_adjusted  \\\n",
       "0        6.291979e-08                2.202193e-07   \n",
       "1        1.092854e-06                3.059992e-06   \n",
       "2        3.351986e-12                1.932321e-11   \n",
       "3        2.513500e-10                1.172967e-09   \n",
       "4        2.536291e-22                2.761739e-21   \n",
       "5        3.016141e-29                4.926364e-28   \n",
       "6        6.511411e-60                3.190591e-58   \n",
       "7        5.718457e-11                2.802044e-10   \n",
       "8        8.438361e-38                2.067398e-36   \n",
       "9        4.004823e-08                1.453603e-07   \n",
       "10       4.004823e-08                1.453603e-07   \n",
       "11       4.004823e-08                1.453603e-07   \n",
       "12       1.978050e-15                1.491145e-14   \n",
       "13       1.978050e-15                1.491145e-14   \n",
       "14       1.978050e-15                1.491145e-14   \n",
       "15       1.279951e-08                4.824430e-08   \n",
       "16       8.262449e-08                2.792138e-07   \n",
       "17       1.802854e-07                5.570495e-07   \n",
       "18       1.818937e-07                5.570495e-07   \n",
       "19       9.181914e-08                2.999425e-07   \n",
       "20       9.181914e-08                2.999425e-07   \n",
       "21       9.181914e-08                2.999425e-07   \n",
       "22       1.372855e-12                8.408736e-12   \n",
       "23       2.884086e-15                2.018860e-14   \n",
       "24       8.800392e-17                7.186986e-16   \n",
       "25       2.077148e-07                6.168500e-07   \n",
       "26       2.726491e-06                7.221517e-06   \n",
       "27       6.419752e-17                5.719416e-16   \n",
       "28       1.211167e-14                7.912957e-14   \n",
       "29       2.580597e-06                7.024960e-06   \n",
       "30       6.668970e-78                6.535591e-76   \n",
       "31       2.336362e-21                2.289635e-20   \n",
       "32       1.085112e-30                2.126819e-29   \n",
       "33       5.550386e-28                7.770540e-27   \n",
       "34       2.359218e-07                6.800099e-07   \n",
       "35       2.359218e-07                6.800099e-07   \n",
       "36       2.359218e-07                6.800099e-07   \n",
       "37       6.265783e-10                2.669768e-09   \n",
       "38       6.265783e-10                2.669768e-09   \n",
       "39       6.265783e-10                2.669768e-09   \n",
       "40       1.128047e-09                4.606191e-09   \n",
       "41       6.055013e-12                3.296618e-11   \n",
       "42       1.937765e-09                7.596038e-09   \n",
       "43       4.692875e-10                2.090462e-09   \n",
       "44       1.047037e-23                1.282620e-22   \n",
       "45       1.047037e-23                1.282620e-22   \n",
       "46       1.047037e-23                1.282620e-22   \n",
       "47       1.202252e-11                6.201092e-11   \n",
       "48       4.436864e-49                1.449376e-47   \n",
       "\n",
       "   Orthology_Correlation_Method Orthology_Regulation_Type Pathway_ID  \\\n",
       "0                      Spearman               Upregulated    ko00430   \n",
       "1                      Spearman               Upregulated    ko00220   \n",
       "2                      Spearman               Upregulated    ko00601   \n",
       "3                      Spearman               Upregulated    ko04024   \n",
       "4                      Spearman               Upregulated    ko00430   \n",
       "5                      Spearman               Upregulated    ko00220   \n",
       "6                      Spearman               Upregulated    ko00601   \n",
       "7                      Spearman               Upregulated    ko00430   \n",
       "8                      Spearman               Upregulated    ko04024   \n",
       "9                      Spearman               Upregulated    ko04151   \n",
       "10                     Spearman               Upregulated    ko05231   \n",
       "11                     Spearman               Upregulated    ko04150   \n",
       "12                     Spearman               Upregulated    ko04530   \n",
       "13                     Spearman               Upregulated    ko04151   \n",
       "14                     Spearman               Upregulated    ko04150   \n",
       "15                     Spearman               Upregulated    ko04151   \n",
       "16                     Spearman               Upregulated    ko00404   \n",
       "17                     Spearman             Downregulated    ko00430   \n",
       "18                     Spearman               Upregulated    ko00430   \n",
       "19                     Spearman               Upregulated    ko04151   \n",
       "20                     Spearman               Upregulated    ko05231   \n",
       "21                     Spearman               Upregulated    ko04150   \n",
       "22                     Spearman               Upregulated    ko04141   \n",
       "23                     Spearman             Downregulated    ko00601   \n",
       "24                     Spearman             Downregulated    ko04141   \n",
       "25                     Spearman             Downregulated    ko00601   \n",
       "26                     Spearman             Downregulated    ko04141   \n",
       "27                     Spearman               Upregulated    ko00430   \n",
       "28                     Spearman               Upregulated    ko00601   \n",
       "29                     Spearman               Upregulated    ko04141   \n",
       "30                     Spearman               Upregulated    ko00430   \n",
       "31                     Spearman               Upregulated    ko00220   \n",
       "32                     Spearman               Upregulated    ko00601   \n",
       "33                     Spearman               Upregulated    ko04024   \n",
       "34                     Spearman               Upregulated    ko04151   \n",
       "35                     Spearman               Upregulated    ko05231   \n",
       "36                     Spearman               Upregulated    ko04150   \n",
       "37                     Spearman               Upregulated    ko04530   \n",
       "38                     Spearman               Upregulated    ko04151   \n",
       "39                     Spearman               Upregulated    ko04150   \n",
       "40                     Spearman               Upregulated    ko04151   \n",
       "41                     Spearman               Upregulated    ko00220   \n",
       "42                     Spearman               Upregulated    ko00430   \n",
       "43                     Spearman               Upregulated    ko04024   \n",
       "44                     Spearman               Upregulated    ko04530   \n",
       "45                     Spearman               Upregulated    ko04151   \n",
       "46                     Spearman               Upregulated    ko04150   \n",
       "47                     Spearman               Upregulated    ko00404   \n",
       "48                     Spearman             Downregulated    ko00430   \n",
       "\n",
       "   Orthology_kegg_no  \n",
       "0             K00485  \n",
       "1             K01476  \n",
       "2             K03371  \n",
       "3             K05208  \n",
       "4             K00485  \n",
       "5             K01476  \n",
       "6             K03371  \n",
       "7             K03852  \n",
       "8             K05208  \n",
       "9             K07207  \n",
       "10            K07207  \n",
       "11            K07207  \n",
       "12            K07298  \n",
       "13            K07298  \n",
       "14            K07298  \n",
       "15            K07825  \n",
       "16            K16438  \n",
       "17            K03119  \n",
       "18            K00485  \n",
       "19            K07207  \n",
       "20            K07207  \n",
       "21            K07207  \n",
       "22            K13999  \n",
       "23            K07966  \n",
       "24            K09523  \n",
       "25            K07966  \n",
       "26            K09523  \n",
       "27            K00485  \n",
       "28            K03371  \n",
       "29            K13999  \n",
       "30            K00485  \n",
       "31            K01476  \n",
       "32            K03371  \n",
       "33            K05208  \n",
       "34            K07207  \n",
       "35            K07207  \n",
       "36            K07207  \n",
       "37            K07298  \n",
       "38            K07298  \n",
       "39            K07298  \n",
       "40            K07825  \n",
       "41            K01476  \n",
       "42            K03852  \n",
       "43            K05208  \n",
       "44            K07298  \n",
       "45            K07298  \n",
       "46            K07298  \n",
       "47            K16438  \n",
       "48            K03119  "
      ]
     },
     "execution_count": 294,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "up_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Taxonomy</th>\n",
       "      <th>Pathway</th>\n",
       "      <th>pathway_kegg_no</th>\n",
       "      <th>Pathway_Correlation</th>\n",
       "      <th>Pathway_p_value</th>\n",
       "      <th>Pathway_p_value_adjusted</th>\n",
       "      <th>Pathway_Correlation_Method</th>\n",
       "      <th>Pathway_Regulation_Type</th>\n",
       "      <th>Orthology</th>\n",
       "      <th>Definition</th>\n",
       "      <th>Orthology_Correlation</th>\n",
       "      <th>Orthology_p_value</th>\n",
       "      <th>Orthology_p_value_adjusted</th>\n",
       "      <th>Orthology_Correlation_Method</th>\n",
       "      <th>Orthology_Regulation_Type</th>\n",
       "      <th>Pathway_ID</th>\n",
       "      <th>Orthology_kegg_no</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Pseudomonas</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K07298</td>\n",
       "      <td>serine/threonine-protein kinase 11</td>\n",
       "      <td>0.251945</td>\n",
       "      <td>2.511328e-08</td>\n",
       "      <td>1.757930e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04530</td>\n",
       "      <td>K07298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Pseudomonas</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K07298</td>\n",
       "      <td>serine/threonine-protein kinase 11</td>\n",
       "      <td>0.251945</td>\n",
       "      <td>2.511328e-08</td>\n",
       "      <td>1.757930e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04151</td>\n",
       "      <td>K07298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Pseudomonas</td>\n",
       "      <td>ko04150</td>\n",
       "      <td>mTOR signaling pathway</td>\n",
       "      <td>0.276996</td>\n",
       "      <td>7.838723e-10</td>\n",
       "      <td>6.270978e-09</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>K07298</td>\n",
       "      <td>serine/threonine-protein kinase 11</td>\n",
       "      <td>0.251945</td>\n",
       "      <td>2.511328e-08</td>\n",
       "      <td>1.757930e-07</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko04150</td>\n",
       "      <td>K07298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Pseudomonas</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K03119</td>\n",
       "      <td>taurine dioxygenase</td>\n",
       "      <td>0.687885</td>\n",
       "      <td>5.730474e-68</td>\n",
       "      <td>1.604533e-66</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>ko00430</td>\n",
       "      <td>K03119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Simonsiella</td>\n",
       "      <td>ko00404</td>\n",
       "      <td>Staurosporine biosynthesis</td>\n",
       "      <td>0.235679</td>\n",
       "      <td>1.973172e-07</td>\n",
       "      <td>1.052358e-06</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>K16438</td>\n",
       "      <td>5-epimerase</td>\n",
       "      <td>0.207119</td>\n",
       "      <td>5.203497e-06</td>\n",
       "      <td>2.913958e-05</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>ko00404</td>\n",
       "      <td>K16438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Simonsiella</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K07966</td>\n",
       "      <td>beta-1,4-galactosyltransferase 1</td>\n",
       "      <td>0.312153</td>\n",
       "      <td>3.230549e-12</td>\n",
       "      <td>3.015179e-11</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>ko00601</td>\n",
       "      <td>K07966</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Simonsiella</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K09523</td>\n",
       "      <td>DnaJ homolog subfamily C member 3</td>\n",
       "      <td>0.356042</td>\n",
       "      <td>1.133038e-15</td>\n",
       "      <td>1.586253e-14</td>\n",
       "      <td>Spearman</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>ko04141</td>\n",
       "      <td>K09523</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Taxonomy  Pathway             pathway_kegg_no  Pathway_Correlation  \\\n",
       "0  Pseudomonas      NaN                         NaN                  NaN   \n",
       "1  Pseudomonas      NaN                         NaN                  NaN   \n",
       "2  Pseudomonas  ko04150      mTOR signaling pathway             0.276996   \n",
       "3  Pseudomonas      NaN                         NaN                  NaN   \n",
       "4  Simonsiella  ko00404  Staurosporine biosynthesis             0.235679   \n",
       "5  Simonsiella      NaN                         NaN                  NaN   \n",
       "6  Simonsiella      NaN                         NaN                  NaN   \n",
       "\n",
       "   Pathway_p_value  Pathway_p_value_adjusted Pathway_Correlation_Method  \\\n",
       "0              NaN                       NaN                        NaN   \n",
       "1              NaN                       NaN                        NaN   \n",
       "2     7.838723e-10              6.270978e-09                   Spearman   \n",
       "3              NaN                       NaN                        NaN   \n",
       "4     1.973172e-07              1.052358e-06                   Spearman   \n",
       "5              NaN                       NaN                        NaN   \n",
       "6              NaN                       NaN                        NaN   \n",
       "\n",
       "  Pathway_Regulation_Type Orthology                          Definition  \\\n",
       "0                     NaN    K07298  serine/threonine-protein kinase 11   \n",
       "1                     NaN    K07298  serine/threonine-protein kinase 11   \n",
       "2           Downregulated    K07298  serine/threonine-protein kinase 11   \n",
       "3                     NaN    K03119                 taurine dioxygenase   \n",
       "4           Downregulated    K16438                         5-epimerase   \n",
       "5                     NaN    K07966    beta-1,4-galactosyltransferase 1   \n",
       "6                     NaN    K09523   DnaJ homolog subfamily C member 3   \n",
       "\n",
       "   Orthology_Correlation  Orthology_p_value  Orthology_p_value_adjusted  \\\n",
       "0               0.251945       2.511328e-08                1.757930e-07   \n",
       "1               0.251945       2.511328e-08                1.757930e-07   \n",
       "2               0.251945       2.511328e-08                1.757930e-07   \n",
       "3               0.687885       5.730474e-68                1.604533e-66   \n",
       "4               0.207119       5.203497e-06                2.913958e-05   \n",
       "5               0.312153       3.230549e-12                3.015179e-11   \n",
       "6               0.356042       1.133038e-15                1.586253e-14   \n",
       "\n",
       "  Orthology_Correlation_Method Orthology_Regulation_Type Pathway_ID  \\\n",
       "0                     Spearman               Upregulated    ko04530   \n",
       "1                     Spearman               Upregulated    ko04151   \n",
       "2                     Spearman               Upregulated    ko04150   \n",
       "3                     Spearman             Downregulated    ko00430   \n",
       "4                     Spearman               Upregulated    ko00404   \n",
       "5                     Spearman             Downregulated    ko00601   \n",
       "6                     Spearman             Downregulated    ko04141   \n",
       "\n",
       "  Orthology_kegg_no  \n",
       "0            K07298  \n",
       "1            K07298  \n",
       "2            K07298  \n",
       "3            K03119  \n",
       "4            K16438  \n",
       "5            K07966  \n",
       "6            K09523  "
      ]
     },
     "execution_count": 295,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "down_results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 기존 Pathway 상관없이"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>KEGG_no</th>\n",
       "      <th>pathway_kegg_no</th>\n",
       "      <th>Control_mean</th>\n",
       "      <th>Cancer_mean</th>\n",
       "      <th>FC_value</th>\n",
       "      <th>Wilcoxon_p</th>\n",
       "      <th>log2FC_value</th>\n",
       "      <th>LogReg_p_univ</th>\n",
       "      <th>LogReg_p_fdr</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>ko04612</td>\n",
       "      <td>Antigen processing and presentation</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.059294e-13</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>ko04926</td>\n",
       "      <td>Relaxin signaling pathway</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>ko00623</td>\n",
       "      <td>Toluene degradation</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.612643e-19</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>ko04320</td>\n",
       "      <td>Dorso-ventral axis formation</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.682350e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>ko04730</td>\n",
       "      <td>Long-term depression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.133567e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>441</th>\n",
       "      <td>441</td>\n",
       "      <td>ko00680</td>\n",
       "      <td>Methane metabolism</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>442</th>\n",
       "      <td>442</td>\n",
       "      <td>ko04970</td>\n",
       "      <td>Salivary secretion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>443</th>\n",
       "      <td>443</td>\n",
       "      <td>ko04016</td>\n",
       "      <td>MAPK signaling pathway - plant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.642694e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>444</th>\n",
       "      <td>444</td>\n",
       "      <td>ko00515</td>\n",
       "      <td>Mannose type O-glycan biosynthesis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>445</th>\n",
       "      <td>445</td>\n",
       "      <td>ko01058</td>\n",
       "      <td>Acridone alkaloid biosynthesis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>446 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Unnamed: 0  KEGG_no                      pathway_kegg_no  Control_mean  \\\n",
       "0             0  ko04612  Antigen processing and presentation           NaN   \n",
       "1             1  ko04926            Relaxin signaling pathway           NaN   \n",
       "2             2  ko00623                  Toluene degradation           NaN   \n",
       "3             3  ko04320         Dorso-ventral axis formation           NaN   \n",
       "4             4  ko04730                 Long-term depression           NaN   \n",
       "..          ...      ...                                  ...           ...   \n",
       "441         441  ko00680                   Methane metabolism           NaN   \n",
       "442         442  ko04970                   Salivary secretion           NaN   \n",
       "443         443  ko04016       MAPK signaling pathway - plant           NaN   \n",
       "444         444  ko00515   Mannose type O-glycan biosynthesis           NaN   \n",
       "445         445  ko01058       Acridone alkaloid biosynthesis           NaN   \n",
       "\n",
       "     Cancer_mean  FC_value  Wilcoxon_p  log2FC_value  LogReg_p_univ  \\\n",
       "0            NaN       NaN         NaN           NaN   6.059294e-13   \n",
       "1            NaN       NaN         NaN           NaN            NaN   \n",
       "2            NaN       NaN         NaN           NaN   1.612643e-19   \n",
       "3            NaN       NaN         NaN           NaN   3.682350e-01   \n",
       "4            NaN       NaN         NaN           NaN   5.133567e-01   \n",
       "..           ...       ...         ...           ...            ...   \n",
       "441          NaN       NaN         NaN           NaN            NaN   \n",
       "442          NaN       NaN         NaN           NaN            NaN   \n",
       "443          NaN       NaN         NaN           NaN   8.642694e-01   \n",
       "444          NaN       NaN         NaN           NaN   1.000000e+00   \n",
       "445          NaN       NaN         NaN           NaN            NaN   \n",
       "\n",
       "     LogReg_p_fdr  \n",
       "0             NaN  \n",
       "1             NaN  \n",
       "2             NaN  \n",
       "3             NaN  \n",
       "4             NaN  \n",
       "..            ...  \n",
       "441           NaN  \n",
       "442           NaN  \n",
       "443           NaN  \n",
       "444           NaN  \n",
       "445           NaN  \n",
       "\n",
       "[446 rows x 10 columns]"
      ]
     },
     "execution_count": 296,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "path_df = pd.read_excel(output_path+'/statistical_analysis_results_with_labels.xlsx')   \n",
    "path_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "anaconda3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
